# Introduction {#intro}

## Cell metabolism ^[a common reference for this section is the textbook @nelson_lehninger_2017]

### Overview

All living organisms and cells must harness non-biological energy, convert it into biological energy and use it to drive all the biological processes needed for their survival, growth, proliferation, and other activities. Biological energy transduction obeys the laws of thermodynamics, including the conservation of energy in an isolated system (i.e. total amount of energy in the universe does not change) and the increase in disorder in an isolated system (i.e. the total entropy of the universe always increases). Under conditions of constant temperature and pressure, which (approximately) applies to most biological systems, the law of entropy increase leads to that for any process to happen spontaneously, the change in Gibbs free energy, $\Delta G$, must be negative. Living organisms and their cells utilizes free energy, i.e. they rely on the decrease in Gibbs free energy to drive their biological activities. While some autotrophs like plants are able to utilize the energy of the radiation from the sun, other organisms including heterotrophs like animals exclusively harness the chemical energy from the molecules they consume in their food and from their environment. By maitaining and carefully regulating tens of thousands of chemical reactions in each cell, they harness the free energy release from the chemical compounds (metabolites) in a highly controlled manner, supporting their normal biological activities and the turn-over of their structural components. This highly complex and fine-tuned network of biochemical processes, the cell metabolism, is fundamental to all life without which no other biologial processes or activity can happen.

Chemical reactions that the cell utilizes can happen spotaneously, i.e. they have negative $\Delta G$, termed thermodynamically unstable. However, although some of the biochemical reactions can readily happen without extra "assistance", most biochemical reactions by themselves happen at very slow rates (i.e. kinetically stable). Therefore for the cell to harness these reactions efficiently, enzymes are needed for most reactions. Enzymes are biocatalysts that significantly increase the rate of chemical reactions. They are usually protein molecules (single or can be large complexes of multiple protein sub-units), which are not consumed in the reactions themselves, but act via decreasing the activation energy of the reaction they catalyze. Different enzymes can be confined within defined cellular compartments, the catalytic activities of each enzyme are usually highly specific to certain reactions, with the function levels/activities of the enzymes subject to swift and highly refined regulations. These mechanisms make it possible for the cell to regulate the entire metabolic network in a precise manner.

The phase of metabolism involving the degradation of larger organic molecules such as carbohydrates, fats, proteins and nucleic acids into smaller products and releasing energy is called catabolism. The chemical energy harnessed during catabolism is often conserved in ATP molecules and reduced electron carriers such as NADH, NADPH and $\mathrm{FADH_2}$. The phase of metabolism involving the biosynthesis of smaller precursor molecules into larger biomolecules is called anabolism. Anabolism requires energy input which is usually in the form of the ATP molecules and reduced electron carriers which are produced during catabolism -- more specifically, the involvement of these "energy-containing" molecules contribute to the spontaneity of the biosynthesis reactions under physiological conditions, allowing them to happen when needed. The cycle of catabolism and anabolism keeps the physical matter of life in constant turn-over accompanied by a constant free energy dissipation and entropy increase, in accordance with the physical laws of thermodynamics.

### Overview of cell metabolic pathways in human

Although then entire cell metabolism has a highly intricate network-like structure, it is conventional to divide the cellular metabolic network conceptually into different parts or sub-systems, called pathways. A common way of pathway definition is based on the type of biomolecules involved, for example, pathways of carbohydrate metabolism, amino acid metabolism, lipid metabolism, nucleic acid metabolism, and so on. Additionally, a central part of cell metabolism responsible for the synthesis of ATP molecules (which acts as the "energy currency" of the cell) is usually termed Energy metabolism, this includes the mitochondrial electron transport chain (ETC)/oxidative phosphorylation (OXPHOS) as well as the TCA cycle which is a central hub for multiple metabolic pathways and responsible for the production of reduced electron carriers which enter the ETC for ATP production. Below I provide a condensed high-level overview of the various key cellular metabolic pathways.

Glycolysis is the process of catabolizing glucose incompletely into pyruvate (via several important intermediate metabolites such as glucose 6-phosphate, 3-phospho-glycerate and phospho-enolpyruvate), producing energy in the forms of ATP and NADH. Depending on whether oxygen is available, pyruvate may be converted into lactate (during which the NADH previously generated is consumed) which is excreted and may be subsequently recycled to produce glucose via gluconeogenesis, or enter the mitochondria and be used for the synthesis of acetyl coenzyme A (Ac-CoA; which then enters the TCA cycle for further catabolism). Gluconeogenesis, on the other hand, is an anabolism pathway for the synthesis of glucose from pyruvate or other glycolytic/TCA cycle intermediates. Via these intermediates, many different types of molecules, including some amino acids and glycerol from triacylglycerols can be used for gluconeogenesis, although mammals including human cannot use fatty acids or Ac-CoA for gluconeogenesis. Gluconeogenesis overall appears to happen in the reverse direction compared to glycolysis but involve several distinct steps of reactions which ensure its irreversibility and differential regulation from glycolysis. Notably, the bypass reaction from pyruvate to phospho-enolpyruvate happens via oxaloacetate and malate in the mitochondria, allowing TCA cycle intermediates to be used for gluconeogenesis. While glucose (together with fructose) is a central metabolic in carbohydrate metabolism, different monosaccharides can be interconverted via specific pathways. Pentose phosphate pathway (PPP) partly can act to interconvert monosaccharides with different numbers of carbon atoms (in the non-oxidative phase of the pathway), but it has another major oxidative phase which serves as an alternative pathway of glucose catabolism. PPP branches from an early upstream point of the glycolysis pathway (glucose 6-phosphate), ultimately oxidizes it completely into $\mathrm{CO_2}$ generating reduced electron carrier NADPH, which can not only be used for various biosynthesis processes but is also important for the cellular antioxidation and redox balance. With incomplete oxidation of glucose in PPP, ribose 5-phosphate is produced which is an important precursor for nucleotide and coenzyme synthesis. Glucose can be used to synthesize large carbohydrate molecules (polysaccharides) like glycogen (in the process of glycogenesis) for storage, and glycogen can also be broken down (glycogenolysis) into glucose; these dynamic interconversion process is regulated by hormones including adrenaline, glucagon and insulin.

As described above, the pyruvate generated in glycolysis can be oxidized to Ac-CoA and $\mathrm{CO_2}$ in the mitochondria, catalyzed by the pyruvate dehydrogenase complex. Ac-CoA, an important activated acetyl group donor, can then enters the TCA cycle for further oxidataion. TCA cycle, occurring in the mitochondrial matrix, is a pivotal cyclic energy metabolism pathway at the center of cell metabolism that can act as a hub connecting multiple pathways of carbohydrate, lipid and amino acid metabolism. The first step of the TCA cycle involves the reaction of Ac-CoA with oxaloacetate, forming citrate, which gives the cycle the other name (citric acid cycle). Within a cycle of about 8 steps involving via citrate, cic-aconitate, isocitrate, $\alpha$-ketoglutarate, succinyl-CoA, succinate, fumarate, and malate, oxaloacetate is regenerated ready to react with another molecule of Ac-CoA thus completing the cycle, while Ac-CoA is being oxidized generating NADH, GTP or ATP, $\mathrm{FADH_2}$, and $\mathrm{CO_2}$. The generated NADH and $\mathrm{FADH_2}$ can further enter the mitochondrial ETC for the synthesis of ATP in OXPHOS. TCA pathway (together with OXPHOS) is a highly effective pathway for oxidizing glucose and generating energy in the form of ATP compared to the incomplete oxidation of glucose via glycolysis. One molecule of glucose when completely oxidized via TCA and OXPHOS can generate 30-32 molecules of ATP, while it can only generate 2 molecules of ATP (and 2 molecules of NADH, which if not consumed in lactate production can further produce about 5 molecules of ATP if entering ETC/OXPHOS) via glycolysis. The many steps and cyclic nature of TCA contributes to its important function as a hub of cell metabolism. The four- and five-carbon metabolites resulted from many different catabolic pathways (e.g. those of amino acids) can be converted into and replenish TCA intermediates via the so-called anaplerotic reactions and thus enter the pathway for oxididation; many TCA intermediates can also be diverted away from the cycle and enter a wide variety of anabolic pathways (e.g. those of carbohydrates, amino acids, fatty acids, porphyrins and heme, etc.). Therefore the TCA cycle is termed an amphibolic pathway, serving both catabolic and anabolic functions.

The reduced electron carriers NADH and $\mathrm{FADH_2}$ generated in the TCA cycle and other cellular catabolic processes enter the mitochondrial ETC, also called respiratory chain, involving several large protein complexes located in the mitochondrial inner membrane, including Complex I (NADH:ubiquinone oxidoreductase), Complex II (succinate dehydrogenase, which is also part of the TCA cycle), Complex III (ubiquinone:cytochrome c oxidoreductase), and Complex IV (cytochrome oxidase). The electrons are passed through these complexes -- from Complex I and II to Complex III via the mobile small molecule carrier ubiquinone (coenzyme Q), and from Complex III to Complex IV via the small protein cytochrome c. Complex IV pass the electron from cytochrome c to molecular oxygen and reduce it to water. Much of the energy released from these reactions are used by Complex I, III and IV, which are also proton pumps, to pump protons from the mitochonrial matrix into the inter-membrane space, creating a gradient in proton concentration. This chemical energe of gradient is utilized by the ATP synthase (Complex V), which also locates in the mitochondrial inner membrane for synthesizing ATP from ADP and inorganic phosphate, coupled to the flow of proton back from the inter-membrane space into the mitochondrial matrix. This process of ATP synthesis, i.e. OXPHOS, is a pivotal metabolic process for generating the energy in the form of ATP that can be used for a wide variety of cellular processes.

Lipid, in the form of triacylglycerols, is break down by lipase into glycerols and free fatty acids. While glycerol can be converted to glycerone 3-phosphate and enters glycolysis for further catabolism, fatty acids can be oxidized to generate large amounts of energy inside the mitochondria in the process called fatty acid oxidation (FAO). Free fatty acids are first activated into acyl-CoA in the cytosol, which is then transported into the mitochondria via the carnitine shuttle. Once inside the mitochondria, the acyl-CoA can undergo $\beta$-oxidation, which breaks down the carbon backbone of the fatty acid in a step-wise fashion (each time a two-carbon unit from the carboxyl end is removed) into Ac-CoA while producing NADH and $\mathrm{FADH_2}$ in every step. Ac-CoA can enter the TCA cycle to produce more NADH and $\mathrm{FADH_2}$, which can enter ETC/OXPHOS to generate large amounts of ATP, which also helps to produce a lot of water. FAO can produce more than two-fold the energy that can be produced by oxidizing the same weight of carbohydrate or protein. Varied reaction steps are needed in $\beta$-oxidation depending on the type of acyl chains (e.g. even or odd number of carbons, saturated or unsaturated, cis or trans configuration of unsaturated chains, etc.). Very-long-chain and branched-chain fatty acids, on the other hand need to be first catabolized into shorter products in the peroxisome via $\beta$-oxidation (which also produces $\mathrm{H_2O_2}$) or $\alpha$-oxidation (for branched-chain fatty acids), then transported to the mitochondria for further oxidation. In the endoplasic reticulum (ER) exists a form of fatty acid oxidation called $\omega$-oxidation that breaks down fatty acid starting from the most distant carbon from the carboxyl group. For lipid anabolism, fatty acid can be biosynthesized from Ac-CoA in the cytosol (in many organisms including human), which is shuttled out of mitochondria as citrate (formed by Ac-CoA reacting with oxaloacetate, and can be converted back to Ac-CoA in the cytosol by citrate lyase consuming ATP). The overall biosynthesis is catalyzed by fatty acid synthase in a step-wise fashion, where the carbon backbone is elongated two carbons a time, similar to FAO. However, each step involves different reactions from those in FAO involving the adding of carbons from the three-carbon intermediate malonyl-CoA while consuming NADPH. The malonyl-CoA required for each step is in turn derived from Ac-CoA (via reacting with bicarbonate catalyzed by Ac-CoA carboxylase, requiring ATP). Fatty acids with longer and more complex chains are synthesized from those with simple and shorter chains in the ER, with the synthesis of unsaturated fatty acids requiring a mixed-function oxidase called fatty acyl-CoA desaturase. Some poly-unsaturated fatty acids cannot be synthesized by mammals but are required precursors for other important products (e.g. eicosanoids), and thus need to be obtained from diet and are termed essential fatty acids. Triacylglycerols can be synthesized by conjugating glycerol 3-phosphate (derived from glycerol or from glycolysis) with acyl-CoAs (derived from fatty acids via the catalysis of acyl-CoA synthetase) though multiple steps. The metabolism of other types of lipids involve distinct pathways. For example, cholesterol can be synthesized from Ac-CoA in the mevalonate pathway (via various important metabolites including HMG-CoA, mevalonate, and squalene) ; cholesterol is the precursor for the further synthesis of many important steroid hormones. Ketone bodies refer to three metabolites including acetone, acetoacetate and $\mathrm{D-\beta}$-hydroxybutyrate. These are derived from Ac-CoA in the liver and can be exported to other organs where they are converted back to Ac-CoA to be used as fuel, especially during starvation where there is increased liver FAO activity to generate Ac-CoA or in diabetes where there is deficiency in Ac-CoA oxidation via TCA due to TCA intermediates being diverted for gluconeogenesis.

Human cells contain 20 common types of amino acids (with additional amino acid derivatives). About 8 of the amino acids (methionine, valine, lysine, leucine, isoleucine, phenylalanine, tryptophan, threonine) cannot be synthesized by the human cell and need to be taken from diet, thus they are termed essential amino acids. Histidine is sometimes considered semi-essential as human body does not always require it. In mammals, tyrosine is derived from the essential amino acid phenylalanine. The rest are called non-essential amino acids. The catabolism of amino acids involves the disposition of the amino group as well as that of the carbon skeletons. Different animo acids can often be interconverted via the process of transamination, where an amino acid and an $\alpha$-keto acid reacts under the catalysis of different transaminases to "swap" their carbon skeleton. Notably, $\alpha$-ketoglutarate can interconvert with glutamate, and oxaloacetate can interconvert with aspartate via this process. This allows the amino group in various amino acids to be transferred to glutamate or aspartate, which can then be disposed of via the urea cycle, while allowing their carbon skeletons to be catabolized as $\alpha$-keto acids, which ultimately enter the TCA for oxidation or be used for other biosynthesis. It needs to be noted that, however, not all amino acids can undergo transamination (exceptions include lysine, threonine and proline), and others like histidine, serine, phenylalaine and methionine are catabolized mainly via other unique pathways rather than via transamination. For urea cycle, which serves to dispose of the amino group, ammonium is first generated in the mitochondria from glutamate via oxidative deamination (requiring $\mathrm{NAD^+}$ or $\mathrm{NADP^+}$) or from glutamine via glutaminolysis. Ammonium joins $\mathrm{HCO_3^-}$ to form carbamoyl phosphate (requiring ATP), which enters the urea cycle by first reacting with ornithine to form citrulline. The rest of the urea cycle happens in the cytosol after citrulline is transported out of the mitochondria, reacting with aspartate to form arginino-succinate, which in turn converts to fumarate and arginine, arginine then breaks down to urea and ornithine, the latter being again imported into the mitochondria to complete the cycle. The urea generated is excreted in human (other species can excrete end products of different forms, e.g. birds excretes the animo group as uric acid). The urea cycle is closely coupled to the TCA cycle via the aspartate-arginino-succinate shunt, where the cytosol aspartate can be derived from oxaloacetate from the TCA cycle which is then exported from the mitochondria, while the fumarate from arginino-succinate breakdown can be converted to malate which then enters the mitochondria to rejoin the TCA cycle. As described above, the TCA cycle is important for the metabolism of the carbon skeleton of amino acids, thus the coupling of TCA cycle and the urea cycle serves to jointly metabolize different chemical parts of amino acids. This leads me to the discussion of the catabolism of amino acid carbon skeletons, which ultimately are mostly either converted to pyruvate or Ac-CoA or enter the TCA cycle at different points. The amino acids that can be catabolized into Ac-CoA (threonine, lysine, leusine, isoleucine, tyrosine, tryptophan, phenylalanine) are termed ketogenic amino acids, as they can then be converted into ketone bodies in the liver. Others that can be converted to the other intermediates are called glucogenic amino acids, as they can be used for gluconeogenesis (some amino acids are both ketogenic and glucogenic, including isoleucine, phenylalanine, tyrosine, tryptophan, and threonine). Specifically, glycine, alanine, serine, threonine, tryptophan and cysteine can be catabolized to pyruvate; proline, glutamate, glutamine, arginine and histidine can be catabolized to $\alpha$-ketoglutarate; isoleucine, methionine, threonine and valine can be catabolized into succinyl-CoA; aspartate and asparagine can be catabolized to oxaloacetate; tyrosine and phenylalanine can be catabolized to fumarate. In human and in other species that are capable of synthesizing all amino acids, diverse pathways centered around glycolysis and the TCA cycle are involved in amino acid biosynthesis: serine is derived from 3-phospho-glycerate in the glycolysis pathway, which can in turn generate glycine and cysteine; alanine, valine and leucine are derived from pyruvate; aromatic amino acids including phenylalanine, tyrosine, tryptophan and histidine are derived via the PPP pathway. Oxaloacetate from the TCA cycle can generate aspartate, which in turn can lead to asparagine, lysine, threonine, isoleucine and methionine. $\alpha$-ketoglutarate from the TCA cycle can generate glutamate, which in turn generates glutamine, proline and arginine (in the urea cycle). Amino acids are used for protein synthesis in the ribosome after conjugated with tRNAs into aminoacyl-tRNAs. Conversely, cellular proteins are under constant turn-over and can be degraded in the proteosome (or lysosome) into amino acids, which can then be reused.

Nucleotides can be synthesized either *de novo* or via salvage pathways which recycle the free bases and nucleosides from nucleic acid breakdown. The *de novo* synthesis of purines involves the production of inosine 5'-monophosphate (IMP) as the first precursor with a purine ring. This process of purine ring building starts from phosphoribosyl pyrophosphate (PRPP, derived from ribose 5-phosphate from the PPP pathway), and requires glycine and two one-carbon unit donated from tetrahydrofolate (THF) for carbons and nitrogens donated from two glutamines and one aspartate. IMP can then be converted to AMP (requiring GTP) and GMP (requiring ATP), which are in turn used to synthesize A/GDP, A/GTP and the corresponding deoxyribonucleotides. The *de novo* synthesis of pyrimidines also requires PRPP leading to the formation of uridine 5'-monophosphate (UMP) as the first pyrimiding-ring containing precursor. The pyrimidine ring is built from aspartate which contributes both carbons and nitrogen atoms, glutamine which contributes another nitrogen atom, and $\mathrm{HCO_3^-}$ which contributes another carbon atom. UMP is in turn used to synthesize UDP and UTP. Various cytosine-containing nucleotides are synthesized starting from CTP converted from UTP, whereas various thymine-containing deoxyribonucleotides are derived from dTMP, converted from dUMP. In general, the conversion of NMPs to NDPs are catalyzed by various nucleoside monophosphate kinases generally specific for a certain base but not the sugar (i.e. ribose or deoxyribose); the conversion of NDPs to NTPs are catalyzed by various nucleoside diphosphate kinases that are generally not specific for either base or the sugar. The generation of deoxyribonucleotides generally happens on the diphosphate level, converting various NDP into dNDP by ribonucleotide reductases. In the salvage pathways, free bases from the breakdown of nucleic acids can react with PRPP to form the corresponding NMP. NTPs are used to synthesize RNA e.g. during transcription via RNA polymerases, while dNTPs are used to synthesize DNA e.g. during DNA replication via DNA polymerases.

I conclude my brief overview of human cell metabolic pathways here, with the note that this is not a complete introduction, with many important aspects of cell metabolism ommitted, including but not limited to the metabolism and function of different vitamins, coenzymes/cofactors, inorganic metabolites like different mineral ions, reactive oxygen species (ROS), reactive nitrogen species, and cellular antioxidation, and the details of the metabolisms of less common types of carbohydrates, lipids, amino acid derivatives and nucleotide/base derivatives. Nevetherless, there is no doubt that cell metabolism underlies all other aspects of cellular activities, it not only supports these other diverse processes but also involves in intricate interactions of double-way regulations with these other domains of cell biology, allowing the optimal response and normal function of cells under a wide variety of conditions.

## Genome-scale metabolic modeling ^[a common reference for many contents throughout this section is the textbook @palsson_systems_2015] 

### Genome-scale metabolic models (GEMs)

GEMs are mathematical network models that contain the information on all cellular metabolic reactions, the enzymes catalyzing the reactions, and the metabolites involved in the reactions and their stoichiometric numbers, allowing the *in silico* modeling and simulation of cell metabolism in a holistic manner using mathematical programming techniques. At its core, GEMs represent the interconnected network of $m$ cellular metabolites and $n$ reactions responsible for their transformation and transportation in a single matrix called the stoichiometric matrix $\mathbf S_{(m\times n)}$, where each row corresponds to a metabolite and each column corresponds to a reaction; an element in the matrix $\mathbf S_{ji}$ is the stoichiometric number of the $j$'th metabolite in the $i$'th reaction (with positive values for products and negative values for reactants). GEM distinguishes different compartments of the cell, including for example the nucleus, mitochondria, endoplasmic reticulum (ER), Golgi apparatus, peroxisome, and extracellular space, among others (depending on the type of species); the same chemical species within different cellular compartments is represented by different metabolites in the model, which are linked by exchange reactions when applicable. The metabolic reactions each has specific limits to their reaction rates (called "fluxes", often denoted by $\mathbf v$) due to thermodynamic and kinetic constraints, given as a vector of lower bound values $\mathbf v^{lb}$ and a vector of upper bound values $\mathbf v^{ub}$ in the model. Non-reversible reactions have flux lower bounds of zeros, whereas reversible reactions have negative flux lower bounds, with the negative flux value representing the reaction occurring in the reversed direction. A GEM also include several special types of "artificial" reactions: Boundary reactions are at the system boundary representing the uptake/secretion of metabolites into/out of the entire metabolic system being modeled, for example a reaction conceptually representing metabolite exchange between the extracellular space and the environment. Sink reaction is a specific type of reversible boundary reaction added to the model to allow certain metabolites to be directly supplied into the metabolic network at certain points (e.g. can be used when the actual reaction producing the metabolite is unknown; it also allows for metabolites being diverted away from the system being modeled). Another specific type of boundary reaction is the demand reaction (or drain reaction), which irreversibly directs metabolites away from the system, conceptually representing the consumption of metabolites for other biologically processes that are not being explicitly modeled. Biomass reaction is a single demand reaction diverting a large range of essential metabolites such as ATP, different amino acids, nucleotides, lipids and carbohydrates away from the modeled system in appropriate ratios corresponding to the specific chemical composition and functionality of the cell being modeled. Conceptually the biomass reaction represents the total metabolic demand for large molecule biosynthesis, maintenance, growth and proliferation of the cell being modeled. Relationships between the genes/enzymes and the reactions, called gene-protein-reaction associations (GPR), are represented by boolean expressions. For example, for a reaction that can be catalyzed by either one of three isozymes A, B and C, the relation is expressed as $A|B|C$, where as if all three genes/proteins D, E and F to form a functional holoenzyme to catalyze a reaction, then the relation is expressed as $D\&E\&F$. A GEM is specifically built for a certain species, or a specific strain (e.g. for unicellular organisms like *E. coli* or yeasts) or a specific tissue/cell type for a multicellular organism like human.

Traditionally, the construction of a GEM takes the so-called bottom-up approach, which involves the long and demanding process of putting together each cellular metabolic reaction, pathway and module in a step-by-step manner, requiring detailed knowledge on the metabolism of the target species or tissue, a large amount of different domains of supporting data and a wide spectrum of different expertises, usually necessitating the collaborative efforts of the research community. The process starts with a draft reconstruction, where all possible metabolic genes/proteins and functions harbored by the target organism/cell are enumerated and collected. This can involve the analysis of the annotated whole genome sequence if the species of interest to indentify all open reading frames (ORFs) that possibly encode metabolic enzymes. The metabolic functions (reactions) of each gene are identified based on existing knowledge, and the GPR associations are established. Transcriptome and other omic data can also be used to utilized to refine the set of metabolic functions relevant to the specific system (e.g. cell/tissue type/organ) of interest. The draft model is then subject to manual curation involving the survey of vast amount of literature to identify more accurate information on the metabolite specificity of the genes/reactions, the chemistry of the metabolites (most basically, their chemical formulae and electric charges), the stoichiometry of the reactions, the thermodynamic/kinetic properties of the reactions (including their reversibility and limits on reaction rates), and the cellular compartments of the metabolites and reactions. Information on the chemical composition, energy demand and growth phenotype of the cell/organism is also needed to create the biomass reaction. Such comprehensively compiled knowledge-base is converted into the computational GEM format. Natually, the necessary information for some genes/metabolites/reactions can be missing, which then accordingly requires experimental studies, and/or theoretical computation/estimates, and the use of computational techniques such as gap-filling in order to produce a functioning model. Due to the prevalence of ambiguities due to incomplete knowledge, there is the practice of assigning confidence scores to different components (e.g. reactions, pathways, etc.) of the constructed model. The constructed model is subject to several evaluations and validations. For example, basic quality checks involve inspecting whether the model can generate all essential precursor metabolites, produce biomass on nutrients/substrates that the cell/organism is known to grow on, and maintain metabolic functions/processes known to the cell/organism of interest (e.g. the secretion of known metabolite products). Further comprehensive validation can be conducted by comparing the model predictions with available experimental data, including (but not restricted to) ATP production efficiency via oxidative phosphorylation (OXPHOS), $\mathrm{O_2}$ consumption and $\mathrm{CO_2}$ production rates, and metabolic gene knockout (KO) phenotypes. The entire process of construction-curation-validation can be iterative to generate the final working model. Many different tools exist to facilitate or automate the particular steps of GEM construction, and many aim to achieve full automation of fast GEM reconstruction [@hamilton_software_2014; @cuevas_dna_2016; @karlsen_automated_2018; @bekiaris_automatic_2020]. These methods involve the automated analysis and annotation of the genome of the species of interest, and takes advantage of databases such as MetaCyc [@caspi_metacyc_2010] and KEGG [@kanehisa_kegg_2012] for prior knowledge on metabolism. An incomplete list of examples include Model SEED [@henry_high-throughput_2010], SuBliMinaL Toolbox [@swainston_subliminal_2011], merlin [@dias_reconstructing_2015], CarveMe [@machado_fast_2018], with benchmark studies comparing the different methods available [@mendoza_systematic_2019]. Softwares have been developed to comprehensively evaluate the quality of constructed models, notably the recent MEMOTE [@Lieven_memote_2020].

Currently, GEMs are available for a wide range of organisms spanning prokaryotes, unicellular and multicellular eukaryotes. GEMs for many classical model species (e.g. the bacteria *E. coli*, the budding yeast *Saccharomyces cerevisiae*, and the house mouse *Mus musculus* [@sigurdsson_detailed_2010; @khodaee_imm1865_2020]), pathogenic organisms (e.g. the gastric bacteria *Helicobacter pylori* and the parasite *Trypanosoma cruzi*), and microorganisms or cell types important for the fermentation/biotechnology industry (e.g. the acetic acid bacteria *Acetobacter pasteurianu* and the Chinese hamster ovary, i.e. CHO cell) are available. Notably, there have been multiple versions of GEMs for human with several iterated updates [@ma_edinburgh_2007; Recon 1, @duarte_global_2007; Recon 2, @thiele_community-driven_2013; Recon 3D, @brunk_recon3d_2018; HMR1, @mardinoglu_integration_2013; HMR2, @mardinoglu_genome-scale_2014; iHsa, @blais_reconciled_2017], with a most recent reconstruction incorporating several previous models [Human1, containing 13,417 reactions, 10,138 metabolites and 3,625 genes, @robinson_atlas_2020] initiating a version-controlled and open-source model development framework based on a GitHub repository, which will potentially accelerate future research by providing real-time rolling updates to the model. Notable online databases hosting large ranges of GEMs include BiGG Models [@king_bigg_2016] and BioModels [@glont_biomodels_2018].

### The methodological framework of GEM

Under ther framework of systems biology modeling with GEM, the fluxes of all metabolic reactions often denoted by the vector $v$ is the variable being modeled. Many different problems in GEM often involve the computation of optimal $\mathbf v$'s or other related objective functions subject to different types of constraints, thus termed constraint-based modeling (CBM). Together with GEM construction, this framework is also termed constraint-based reconstruction and analysis (COBRA). Different mathematical optimization techniques are used in CBM/COBRA, such as linear programming (LP), quadratic programming (QP), mixed integer programming (MIP), and non-linear programming (NLP).

The most important set of constraints making the optimization problems solvable in CBM/COBRA is that imposed by the steady state assumption, which assumes that the metabolic system being modeled is in a steady state with no net change in the concentration of any metabolite. Essentially, this assumption requires that the total production flux of any one metabolite equals to its total consumption flux, thus there is no net accumulation or depletion of any metabolite within the system. Mathematically, this corresponds to the set of linear constraints expressed by $\mathbf{Sv=0}$, also termed flux balance. Such an assumption is apparently seldom strictly satisfied, as it is rare, if not impossible to find a real-life biological system to be strictly under steady state. However, the steady state assumption is a quite reasonable one under biological homeostasis and many experimental conditions where the cells or system under study is in a stable biological state over a reasonably long duration of time. Besides, many biological systems can be regarded as being in quasi-steady states, which means that a system with relatively fast kinetics can be practically in steady states although its environment is changing with dynamic inputs and outputs between the system and the environment. The steady state assumption and flux balance are central and highly useful in CBM/COBRA, as without this assumption, the system would have to be modeled by kinetic modeling involving the kinetic constants of the reactions, which in many cases are unknown, rendering the system impractical to analyze. In practise, flux balance is usually assumed for all metabolites in the model, although if certain metabolites are known to not satisfy the assumption and their fluxes of net accumulation/consumption are known, the flux balance constraints for these metabolites can be replaced by the corresponding constraints on flux production or consumption as appropriate.

The flux balance constraints restricts the viable solution space of a model comprising $n$ reactions from $\mathbb R^n$ to a subspace of $\mathbb R^n$, or specifically a hyperplane. Imposing the lower and upper bounds of the reaction fluxes further shrinks the space, i.e. one defined as:

$$
\begin{aligned}
&\mathbf{Sv = 0} \\
&\mathbf{v^{lb} \le v \le v^{ub}} \\
\end{aligned}
$$

When the reactions in the model are defined in a way that all fluxe values are positive (i.e. each reversible reaction $i$ is expressed as two "half reactions" whose flux lower bounds are zero and upper bounds are $|v^{lb}_i|$ and $|v^{ub}_i|$, respectively), then this solution space is a bounded subset of a convex cone. Additional constraints based on reaction kinetics and thermodynamics, metabolite concentrations, gene and protein expression, etc. can be formulated and help to further shrink the solution space, although they cannot shrink the space to a single point, namely, a single unique solution generally cannot be found only by imposing the various constraints. Therefore, many different optimization algorithms are applied to obtain a unique or at least more restricted set of solutions for detailed analysis, depending on different biologically informed assumptions, and additional techniques including sampling can be used to analyze a non-uniquely defined space of solutions.

One most basic and important optimization method in GEM to obtain unique or a more restricted set of solutions is the flux balance analysis (FBA), where certain cellular objective function is maximized (or minimized). A cellular objective function in FBA $f_{obj}(\mathbf{v})$ is a single scalar-valued linear function of the reaction flux vector $\mathbf{v}$ representing a certain biological process or function, a common example being the biomass reaction -- In many cases, for example cultured bacteria or other actively proliferating cells in optimal (rich) growth media, it is reasonable to assume that the system being modeled is under optimal growth, represented by maximal biomass production. Maximizing the biomass production is formulated as the linear programming (LP) problem below:

$$
\begin{aligned}
&\text{maximize } f_{obj}(\mathbf{v}) = v_{\text{biomass}} \\
&\text{subject to:} \\
&\mathbf{Sv = 0} \\
&\mathbf{v^{lb} \le v \le v^{ub}} \\
\end{aligned}
$$

Similarly, other cellular objective functions can be specified as appropriate depending on the prior knowledge of the cell type or organism under study. FBA can also be used to study the capacities of various biological functions under different conditions by specifying the corresponding cellular objective functions, e.g. co-factors (e.g. NADH or NADPH), biosynthetic precursors or ATP production. Different biological conditions can be specified with additional constraints, for example, nutrient/metabolite availability can be specified by the flux bounds of their boundary reactions, and metabolic gene/reaction knockdowns or knockouts can be represented by decreased or zero flux bound values of the corresponding reaction. In all cases, since both the objective function and the sets of constraints are linear, the problems can be formulated as LP problems which can be effectively solved by well-known algorithms such as the simplex algorithm.

Although the bounded LP problem in FBA is guaranteed to identify the optimal objective value, the optimal solutions $\mathbf{v^{opt}}$ that can yield the optimal objective value may be non-unique, in which case they are termed degenerate solutions. Such cases are very common when modeling a large genome-scale model. Many techniques exist for further analysis of the optimal solution(s) of FBA and the space of degenerate solutions. For example, the sensitivity of the optimal solution can be analyzed by shadow prices and reduced costs. The shadow price of a boundary reaction is the incremental change in the optimal objective function value given the incremental flux change of the boundary reaction (i.e. the derivative of the former with respect to the latter), i.e. it measures whether or how the optimal objective value is affected by the availability of a certain metabolite. Similarly, the reduced cost is defined for an internal reaction (i.e. not boundary) and measures whether or how the change of an internal flux affects the optimal objective value. The presense of metabolites/reactions with zero shadow price/reduced cost implies the existence of degenerative solutions. These analyses of the effect of varying one parameter (the flux of the $I$'th reaction) on the optimal objective value, called robustness analysis, are conducted via solving a series of LP problems with different fixed $\theta$ values:

$$
\begin{aligned}
&\text{maximize } f_{obj}(\mathbf{v}) \\
&\text{subject to:} \\
&\mathbf{Sv = 0} \\
&v_I = v_{I,min} + \theta (v_{I,max}-v_{I,min}),\; \theta \in [0,1]\\
&v^{lb}_{i} \le v_i \le v^{ub}_{i}\, ,\; i=1,...,n,\, i\neq I\\
\end{aligned}
$$

Similarly, the effect of varying two parameters on the optimal objective function value can be analyzed, which produces a 2D surface called phenotypic phase plane. Another more direct way of analyzing degenerate solutions is the flux variability analysis (FVA), which computes the range of possible fluxes of each reaction while fixing the cellular objective function at its optimal value $f_{obj}^{opt}$. FVA is formulated as follows:

$$
\begin{aligned}
&\text{maximize and minimize } v_{i},\; \text{for each } i=1,...,n \\
&\text{subject to:} \\
&f_{obj}(\mathbf{v}) = f_{obj}^{opt}\\
&\mathbf{Sv = 0} \\
&\mathbf{v^{lb} \le v \le v^{ub}} \\
\end{aligned}
$$

Besides, the space of degenerate solutions, or any convex space of solutions in LP problems can in theory be characterized using uniform random sampling, i.e. sampling of a statistically sufficient number of solutions that lie uniformly within the solution space. These sample points reflecting the flux distribution under the optimal objective value can then be analyzed and visualized to study the variability in the metabolic state under the optimal cellular objective function. Sampling a high-dimensional space, however, can be a challenging problem, especially that the solution space of GEM can often have irregular and elongated shape. The sampling of large GEMs can also take a very long time to converge. A well-known sampling algorithm suitable for the sampling of high-dimensional spaces with irregular shapes is called artificial centering hit-and-run (ACHR), and many variations of the algorithm have been developed [@kaufman_direction_1998; @megchelenbrink_optgpsampler_2014; @saa_ll-achrb_2016]. Coordinate hit-and-run with rounding (CHRR) is another efficient algorithm shown to have fast convergence time [@haraldsdottir_chrr_2017; @fallahi_comparison_2020]. Rather than sampling, a recent algorithm based on expectation propagation was shown to accurately approximate the marginal distribution of high-dimensional polytopes with high computation efficiency [@braunstein_analytic_2017]. These algorithms are valuable for more detailed analyses of the convex solution spaces arising from the CBM/COBRA of GEMs.

### GEM algorithms for omic data integration

GEM modeling has proven invaluable for the *in silico* simulation of the metabolism of many different organisms, producing results with good concordance with experimental data. For unicellular organisms or cells of multicellular organisms under culture, the alteration in cell metabolism under different culture conditions can be readily modeled, provided that a GEM tailored for the cell type under study has been constructed. However, for complex multicellular organisms like human, a large number of different tissue and cell types exist with highly diverse and distinct metabolic functions, in theory requiring a customized GEM for each. Furthermore, for the modeling of *in vivo* metabolism, the cells of different organs and tissues are within different metabolic environments that are not pre-defined. Variabilities across conditions or individuals due to other factors only indirectly related to cell metabolism are also not as straightforward to model as, say, changes in metabolite availability in the culture media or direct perturbation of metabolic genes/enzymes. Fortunately, the advent and fast progress made in different omic profiling technologies has allowed the fine-grained and high-throughput characterization of cellular molecular landscapes in any biological samples under any experimental conditions. Whole-transcriptome gene expression data from microarrays or various RNA-sequencing technologies, high-throughput protein expression data from mass spectrometry-based quantitative proteomics, and metabolomics data that quantifies the level of large numbers of metabolites in a sample are now widely available. The integration of such omic data into GEM has truly unleashed the power of GEM modeling for the study of cell metabolism in a greater range of problems. The various omic data which provides detailed high-coverage molecular profiles is often analyzed with statistical modeling techniques, which can reveal complex patterns in the data but often cannot easily establish any causal relationships. In contrast, the CBM/COBRA framework of GEM modeling, due to its capacity to perform mechanism-based *in silico* simulations, can be used to investigate causality. Thus, the integration of GEM with omic data provides a valuable opportunity to deepen the understanding of the molecular system under study, specifically within the domain of metabolism.

Next I will focus on the discussion of integration of gene (or protein) expression data into GEM. A large number of different algorithms exist for this important problem [@kim_methods_2014; @opdam_systematic_2017; @jamialahmadi_benchmark-driven_2019]. The common idea underlying these algorithms is to either select a subset of high-confidence reactions from the general GEM ("base model") of an organism that are supported by the additional omic data, and/or to adjust the flux bounds of the reactions or directly fit the flux values based on the notion of a rough correlation between gene/protein expression levels and the activity levels (i.e. fluxes) of the corresponding reactions, resulting in refined GEMs and/or optimal flux distributions specific to the sample or context that correspond to the external omic data. Some of the algorithms of the first type include MBA [@jerby_computational_2010], mCADRE [@wang_reconstruction_2012], FASTCORE [@vlassis_fast_2014], FASTCORMICS [@pacheco_integrated_2015], and CORDA [@schultz_reconstruction_2016], which involves the identification of "core" sets of reactions supported by external data including expression data, supplemented usually in a parsimonious fashion by additional reactions necessary to yield a consistent model without "dead-end" reactions that cannot bear fluxes. These algorithms generates customized GEMs that correspond to the external data, but does not directly compute an optimal flux distribution. Most other algorithms are of the second type, which seek to maximize the concordance of flux and gene expression in various ways. GIMME [@becker_context-specific_2008], E-Flux [@colijn_interpreting_2009], Lee et al. [@lee_improving_2012], $\text{GIM}^3\text{E}$ [@schmidt_gim3e_2013], E-Fmin [@song_prediction_2014], and TRFBA [@trfba_motamedian_2017] each applies different LP-based formulations to maximize the concordance between the fluxes and gene expression values (after converted to the reaction level based on the GPR boolean mapping). Also based on this idea but instead of using LP-based formulations, iMAT [@shlomi_network-based_2008], INIT [@agren_reconstruction_2012] and tINIT [@agren_identification_2014] use MILP-based formulations to maximize the number of reactions with high or low fluxes that are consistent with the corresponding high or low gene expression levels (defined based on pre-defined thresholds as parameters). RegrEx [@estevez_context-specific_2015] is based on a MIQP formulation that minimizes the difference between flux and gene expression data with $\ell 1$-norm regularization. PRIME [@yizhak_phenotype-based_2014] requires the gene expression data of multiple samples together with the data on measured cell growth rates of the samples, which are used to identify a subset of reactions correlating with the measured cell growth. The flux bounds of these reactions are then directly adjusted according to the gene expression values. Many additional algorithms are available [e.g. @zhang_rmetd2_2019], however, few attempt has been made to compare and benchmark the various algorithms in a comprehensive manner [@opdam_systematic_2017; @jamialahmadi_benchmark-driven_2019], and the available benchmark studies only cover a subset of the available methods. The benchmark studies mostly adopted technical evaluations (e.g. robustness to noisy data) and indirect methods (e.g. predicting known metabolic functionalities, gene essentiality, drug response, etc.) to validate the performance of the algorithms, as large-scale data of direct flux measurements is very rare. Different algorithms can each involve numerous parameters whose varied values can affect the performance of the algorithms. For example, many algorithms requires a pre-determined gene expression level threshold to define high vs low expression, and the benchmark study of Opdam et al. that compared MBA, mCADRE, FASTCORE, iMAT, INIT and GIMME [@opdam_systematic_2017] shows that in some cases, the selection of this threshold can contribute to more variation in the whether a reaction is retained in the resulting model even compared to the choice of algorithm; In gene essentiality prediction, however, the most variation appears to arise from the choice of algorithm, and using stricter gene expression thresholds as well as properly constraining the GEM based on culture media composition and exometabolomic data overall helped to improve the prediction accuracy; mCADRE and MBA seemed to performed the best in the gene essentiality prediction, followed by INIT and iMAT (while these two showing high variation in their performances depending on dataset and parameters used); GIMME and FASTCORE performed worse than the other algorithms in this task. The constraining of the model based on metabolite uptake also significantly affected growth rate prediction. The study concluded that no single algorithm is a universal best performer and algorithms need to be chosen based on the available data and goal of the study [@opdam_systematic_2017]. The benchmark study by Jamialahmadi et al. [@jamialahmadi_benchmark-driven_2019] compared mCADRE, FASTCORE, FASTCORMICS, CORDA, iMAT, INIT, PRIME, GIMME, and TRFBA based on technical assessments and a few tasks specifically based on cancer cell lines including predicting cell growth rate, metabolites uptake/secretion rates, drug response and gene essentiality, and cancer driver gene prediction. This study confirms the findings in the Opdam et al. study that the algorithm performance depends on the specific task, and constraining metabolic uptake based on culture media helped to improve performance notably for growth rate prediction, although constraint models did not improve essential gene prediction much except for TRFBA. CORDA and iMAT was found to perform worse than the other algorithms in predicting growth rate. PRIME was the only algorithm found to be able to produce significant predictions on metabolite uptake/secretion rates without constraining the model based on the media, but once with proper constraining, other algorithms including GIMME, TRFBA and simply parsimonious FBA (pFBA, without expression data integration) can produce better performances. PRIME and TRFBA appeared to perform the best on drug sensitivity and gene essentiality prediction, however, both algorithms takes advantage of data on measured growth rates. In gene essentiality prediction, GIMME also had reasonable performance, but mCADRE performed worse, which is contradictory to the Opdam et al. results potentially due to the variation in the dataset and detailed algorithm parameters used, highlighting the limitations of current benchmark studies. On the other hand, FASTCORE, FASTCORMICS, INIT, and iMAT had good performances in various technical assessments including robustness to noise, lack of blocked reactions and similarity in the obtained model given similar input data. The authors further proposed a new algorithm combining TRFBA and FASTCORMICS which they called TRFBA-CORE, which does not require measured growth rate data, and showed it to have overall the most balanced performance across multiple tasks and technical evaluations. Despite this, the authors highlighted the need for further more robust benchmarks and the choice of method based on the problem under study.

As one example of the many gene expression data integration algorithms, and also due to the specific relevance to the work in this thesis, I next describe the Integrative Metabolic Analysis Tool (iMAT) algorithm [@shlomi_network-based_2008] in more detail. iMAT takes the transcriptome-wide gene expression data of a sample in a vector, together with the "base" GEM of the corresponding organism as input; it outputs a GEM customized to the given sample, with reaction flux bounds further constrained to fit that of the expression profile. The input vector contains the within-sample-normalized expression values of all genes in a sample (i.e. the values should ideally be comparable between any two genes), two threshold parameters are used to divide all genes into three sets: high-expression, medium-expression, and low-expression. Then the gene expression vector is mapped to a discretized vector of $\{1,0,-1\}$, with high, medium, and low-expression genes mapped to 1, 0, and -1, respectively. This vector of discretized gene expression values is then converted to a vector of expected reaction activity levels, based on the boolean GPR association, but replacing the logical $\&$ and $|$ operators with $max$ and $min$ functions, respectively. The resulting reactions with values of 1 are expected to have high activity, denoted $R_H$, and those with values of -1 are expected to have low activity, denoted $R_L$. Only $R_H$ and $R_L$ are used in the subsequent optimization procedure, and reactions with values of 0 are not used. The subsequent optimization seeks to maximize the number of reactions bearing consistent flux levels with those expected from the expression data, while subject to the usual constraints of the metabolic network (i.e. flux balance and the flux bounds). This is formulated as an MILP problem by introducing a series of integer (actually binary) variables $y^+_i$'s and $y^-_i$'s, for each reaction $i\in R_H$ one $y^+_i$ and one $y^-_i$ is introduced, and for each reaction $i\in R_L$ only one $y^+_i$ is introduced. The formulation is such that for a reaction $i\in R_H$, $y^+_i=1$ corresponds to that it has large positive flux greater than the positive parameter $\epsilon_{hi}$ (i.e. high activity level in the positive direction), $y^-_i=1$ corresponds to large negative flux (smaller than $-\epsilon_{hi}$), and $y^+_i=y^-_i=0$ corresponds to the default flux bound constraint in the "base" model; obviously $y^+_i$ and $y^-_i$ cannot both be 1 for $i\in R_H$; for a reaction $i\in R_L$, $y^+_i=1$ corresponds to that it has small flux value near zero (absolute flux value smaller than the positive parameter $\epsilon_{lo}$), and $y^+_i=0$ corresponds to the default flux bound constraint:

$$
\begin{aligned}
&\maximize_{\mathbf{v,y^+,y^-}} \bigg( \sum_{i\in R_H}(y^+_i+y^-_i) + \sum_{i\in R_L}y^+_i \bigg)  \\
&\text{subject to:} \\
&\mathbf{Sv = 0} \\
&\mathbf{v^{lb} \le v \le v^{ub}} \\
&v_i + y^+_i(v^{lb}_i - \epsilon_{hi}) \ge v^{lb}_i,\; \text{for } i\in R_H \\
&v_i + y^-_i(v^{ub}_i + \epsilon_{hi}) \le v^{ub}_i,\; \text{for } i\in R_H \\
&v_i + y^+_i(v^{lb}_i + \epsilon_{lo}) \ge v^{lb}_i,\; \text{for } i\in R_L \\
&v_i + y^+_i(v^{ub}_i - \epsilon_{lo}) \le v^{ub}_i,\; \text{for } i\in R_L \\
&y^+_i,\, y^-_i \in \{0,1\} \\
\end{aligned}
$$

The original formulation of iMAT as in [@shlomi_network-based_2008] set the parameter $\epsilon_{hi}$ to 1 and does not have the $\epsilon_{lo}$ parameter, and is equivalent to fixing $\epsilon_{lo}$ at zero. After solving the MILP problem, $v^{lb}_i,\; i\in R_H \land y^{+,opt}_i=1$ are set to $\epsilon_{hi}$, $v^{ub}_i,\; i\in R_H \land y^{-,opt}_i=1$ are set to $-\epsilon_{hi}$, and $v^{ub}_i,\; i\in R_L \land y^{+,opt}_i=1$ are set to $\epsilon_{lo}$, $v^{lb}_i,\; i\in R_L \land y^{+,opt}_i=1$ are set to $max(-\epsilon_{lo}, v^{lb}_i)$. To address the issue of possibly multiple optimal solutions, The above MILP problem is run twice for each reaction, one time forcing the reaction to be activated (having flux greater than $\epsilon_{hi}$) and the second time forcing it to be inactivated (having flux lesser than $\epsilon_{lo}$), the resulting optimal objective function values of the two runs, $f^{obj,act}_i$ and $f^{obj,inact}_i$ are compared, and if $f^{obj,act}_i > f^{obj,inact}_i$, reaction $i$ is determined to be activated, if $f^{obj,act}_i < f^{obj,inact}_i$, reaction $i$ is determined to be inactivated, if $f^{obj,act}_i = f^{obj,inact}_i$, the activity level of reaction $i$ is undetermined and the corresponding flux bounds are not changed (left as default as in the base model). Alternatively, analogous to the FVA procedure, after solving the base MILP, an additional linear constraint can be added forcing the objective function at its optimal value, then a series of MILPs can be solved to determine the maximal and minimal fluxes of each reaction.

### Additional GEM algorithms relevant to this thesis

Many GEM algorithms have been developed for various specific purposes. In this section I describe a few additional algorithms that are relevant to the work in this dissertation.

Minimization of Metabolic Adjustment (MOMA) [@segre_analysis_2002] is an algorithm to predict the flux distribution after a metabolic perturbation, i.e. a metabolic reaction knockout (KO). Although this prediction can be also achieved with FBA after simulating the KO by constraining the corresponding reaction flux to zero (or a very low value), MOMA is based on the idea that after the KO, the preference of the system in the short term is to make only minimal adjustments and shift to a nearest point in the new feasible metabolic space. MOMA adopts the Euclidean distance measure to determine the "nearest point", leading to a QP problem formulated as below (where $\mathbf{v^{opt,wt}}$ is the optimal solution obtained from FBA on the wild-type (WT) model without any KO, and $v_{i} = 0$ simulates the KO of the $i$'th reaction):

$$
\begin{aligned}
&\minimize_{\mathbf{v}} ||\mathbf{v}-\mathbf{v^{opt,wt}}||_2^2 \\
&\text{subject to:} \\
&\mathbf{Sv = 0} \\
&\mathbf{v^{lb} \le v \le v^{ub}} \\
&v_{i} = 0
\end{aligned}
$$

However, to address the issue of possible degenerate solutions in the FBA, an alternative QP formulation is given below, where the WT model and the KO model are optimized together ($f_{obj}^{opt,wt}$ is the optimal cellular objective function value obtained from FBA in the WT model):

$$
\begin{aligned}
&\minimize_{\mathbf{v^{wt},v^{ko}}} ||\mathbf{v^{wt}}-\mathbf{v^{ko}}||_2^2 \\
&\text{subject to:} \\
&f_{obj}(\mathbf{v^{wt}}) = f_{obj}^{opt,wt}\\
&\mathbf{Sv^{wt} = 0} \\
&\mathbf{Sv^{ko} = 0} \\
&\mathbf{v^{lb,wt} \le v^{wt} \le v^{ub,wt}} \\
&\mathbf{v^{lb,ko} \le v^{ko} \le v^{ub,ko}} \\
&v^{ko}_{i} = 0
\end{aligned}
$$

Metabolic Transformation Algorithm (MTA) [@yizhak_model-based_2013] is an algorithm for the prediction of metabolic targets whose KO can transform the cellular metabolic state from a given reference state to a given target state. For example, MTA can be used to predict therapeutic targets for reversing the disrupted cell metabolism in a metabolism-related disease. The input to MTA is the transcriptome-wide gene expression data of the reference state, in addition to the differential gene expression (DE) changes in the target state compared to the reference state, provided as two vectors; MTA outputs a score called the MTA score for each reaction, with higher MTA scores corresponding to better targets, i.e. whose KO can better transform the cell metabolism from the reference to the target state. A first step of MTA involves applying the iMAT algorithm to obtain a GEM customized to the reference state. Then uniform random sampling is performed on the reference GEM, and the mean vector $\mathbf{v^{ref}}$ of all sample points is obtained as the reference metabolic flux distribution of the reference state. Up-regulated, down-regulated and stable genes are then determined from the DE results, which are mapped based on the boolean GPR associations to the reaction level to obtain the sets of reactions that are expected to have increased activity level (i.e. absolute flux) or decrease activity level. Specifically, if a reaction is expected to have increased activity level if it is catalyzed by a set of enzymes with $\&$ relations and all of the corresponding genes are up-regulated, or if it is catalyzed by a set of enzymes with $|$ relations and at least on of the corresponding genes are up-regulated; similarly reactions expected to has decreased activities are determined. All other cases not covered in the above description are regarded as undertermined. For the reactions with expected increase in activity and whose fluxes are positive in the reference state $\mathbf{v^{ref}}$, they are expected to have further increased flux value (i.e. further flux change in the forward direction); similarly reactions whose flux in $\mathbf{v^{ref}}$ is negative and expected to have decrease in activity level also corresponds to expected flux change in the forward direction. The set of these reactions is denoted $R_F$. Similarly the set of reactions whose flux value are expected to change in the backward/reverse direction can be identified and is denoted $R_B$. Any undetermined reaction is regarded as haveing stable flux values and the corresponding set is denoted $R_S$. Next, a series of MIQP problems are solved to maximize the number of reactions with flux changes concordant with the expectations, while minimizing the squared sum of flux changes in reactions that are expected to have stable fluxes, under the simulated KO of each reaction $j$:

$$
\begin{aligned}
&\minimize_{\mathbf{v,y}} \bigg( (1-\alpha)\sum_{i\in R_S}(v^{ref}_i-v_i)^2 + \frac{\alpha}{2} \sum_{i\in R_F \cup R_B }y_i \bigg)  \\
&\text{subject to:} \\
&\mathbf{Sv = 0} \\
&\mathbf{v^{lb} \le v \le v^{ub}} \\
&v_j=0 \\
&v_i - y^F_i(v^{ref}_i + \epsilon_i) - y_i v^{lb}_i \ge 0,\; \text{for } i\in R_F \\
&y^F_i + y_i = 1,\; i\in R_F \\
&v_i - y^B_i(v^{ref}_i - \epsilon_i) - y_i v^{ub}_i \le 0,\; \text{for } i\in R_B \\
&y^B_i + y_i = 1,\; i\in R_B \\
&y_i,\, y^F_i,\, y^B_i \in \{0,1\} \\
\end{aligned}
$$

In the formulation above, for each reaction $i\in R_F$, binary variables $y_i$ and $y^F_i$ are introduced such that one and only one of the two variables can be 1 (while the other has to be 0); similarly binary variables $y_i$ and $y^B_i$ are introduced for each reaction $i\in R_B$. $y^F_i=1$ or $y^B_i=1$ correspond to "successful" flux change that is consistent with the expected changes for reaction $i$, and thus $y_i=1$ means a failure to achieve the expected flux changes. Therefore, the integer part of the minimization problem corresponds to minimizing the number of failed flux changes. The relative weights of this part and the quadratic part (which minimizes the squared sum of flux changes in reactions that are expected to have stable fluxes) are determined by the parameter $\alpha$. In the original study [@yizhak_model-based_2013], performance was found to be robust to the variation in this parameter. $\epsilon_i$ is a parameter that corresponds to the size of flux change used to define a "successfully" changed reaction, while it can be reaction-specific in principle, it was shown that a fixed value, e.g. 0.01 can achieve reasonable and equivalently good performance.

After solving the above MIQP problem for the KO of each reaction, the optimal solution is obtained and thereby the sets of "successfully" changed reactions as well those failed to achieve the expected change can be identified. The MTA score for the KO is then computed as follows:

$$
\sum_{i\in R_{success}}|v^{ref}_i-v^{opt}_i| - \sum_{i\in R_{fail}}|v^{ref}_i-v^{opt}_i| \over \sum_{i\in R_S}|v^{ref}_i-v^{opt}_i|
$$

The MIQP problem can also be solved using the WT model without any KO, which can also produce an MTA score that can be used as a control. Only reactions whose scores are greater than the control scores are potentially useful targets. Based on the validations reported in [@yizhak_model-based_2013], the 10%-20% reactions with the highest MTA scores appear to represent biologically meaningful targets in that in most cases of validation datasets, the reaction corresponding to the actual gene KO lies within top 10%-20% MTA predictions.

## Metabolism of T cells in cancer and cancer immunotherapy

### Overview of cancer immunology



### Cancer immunotherapy and factors associated with patient response



### Immunometabolism in T cell differentiation and function



## Metabolism in COVID-19 and host-SARS-CoV-2 interaction

### COVID-19: natural history, epidemiology and clinical features 

The coronavirus disease 2019 (COVID-19) is a severe and highly infectious respiratory disease whose first known case appeared in late December 2019 in Wuhan, China [@zhu_novel_2020]. Since then, COVID-19 has quickly evolved into a global pandemic within a few months by March 2020 [@who_who_2020] and is still ongoing, resulting in more than 100 million infected cases and over 2 million deaths (as of March 2021) [@who_who_2020-1]. The pathogen for COVID-19 has been identified to be a new betacoronavirus that was never discovered before, and was named severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) [@coronaviridae_study_group_of_the_international_committee_on_taxonomy_of_viruses_species_2020]. Two related viruses of the coronavirus family (*Coronaviridae*), the severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle East respiratory syndrome coronavirus (MERS-CoV), emerged in 2002 and 2012 respectively have also caused serious respiratory diseases [@cui_origin_2019], making COVID-19 the third coronavirus-related severe infectious disease discovered in the recent decades. Like SARS-CoV and MERS-CoV [@cui_origin_2019], SARS-CoV-2 is likely to be of zoonotic origin, with the first COVID-19 cases epidemiologically linked to a wet market in Wuhan selling seafood, live poultry and wildlife [@jiang_emerging_2020]. Phylogenetic studies have identified high similarities between the genomic sequences of SARS-CoV-2 and a group of betacoronaviruses of the subgenus *Sarbecovirus* previously found in bats in China [@wu_new_2020; @zhou_pneumonia_2020], and SARS-CoV-2-related coronaviruses have also been found in Malayan pangolins (*Manis javanica*) [@lam_identifying_2020]. However, the identity of the direct progenitor of SARS-CoV-2 is still uncertain [@andersen_proximal_2020], and the intermediate host responsible for the zoonotic transmission of the virus from the wild reservoir host species to human has not been confidently identified [@zhao_potential_2020].

Airborne transmission has been determined to be the dominant route of transmission of COVID-19 [@zhang_identifying_2020], specifically via respiratory droplets and potentially aerosol [@van_doremalen_aerosol_2020], fecal-oral transmission can also be possible [@xu_characteristics_2020]. COVID-19 is highly transmissible, with a basic reproduction number (the expected number of infections directly caused by transmission from one primary infected individual when all individuals are susceptible) estimated to be around 2 or 3 [@zhao_preliminary_2020; @billah_reproductive_2020], similar to SARS and higher than e.g. the 2009 influenza and MERS [@petersen_comparing_2020]. The incubation period (time from infection to symptom onset) of COVID-19 was estimated to range from 1.3 to 11.3 days [@lauer_incubation_2020], longer than SARS [@petersen_comparing_2020]. SARS-CoV-2 show a distinct kinetics of virus shedding, with high viral load when symptoms begin and a subsequent decline but persisting for a long period, making it critical to timely isolate and quarantine patients as soon as infection is detected and for sufficiently long time [@wolfel_virological_2020; @zheng_viral_2020; @petersen_comparing_2020]. COVID-19 is less deadly than SARS or MERS, with possibly more than 50% infected cases being asymptomatic [@li_asymptomatic_2020], and most symptomatic cases being mild [@wu_characteristics_2020]. However, asymptomatic patients can still spread the virus [@li_asymptomatic_2020], adding to the challenge of transmission control. The infection-fatality risk was estimated to be 1.39% in the New York City, the first epidemic center in the USA [@yang_estimating_2021], and other estimates for other regions in the world were mostly below 1% [@streeck_infection_2020; @pastor-barriuso_infection_2020]. Older age and various comorbidities including hypertension, obesity, diabetes and coronary heart disease (among others) were identified as the risk factors for SARS-CoV-2 infection and/or severe disease [@mao_assessing_2020; @jordan_who_2020].

The most common initial symptoms of COVID-19 include dry cough, fever, and fatigue, additional symptoms can include sputum production, sore throat, haemoptysis, chills, headache, chest pain, anorexia, diarrhea, and nausea/vomiting [@hu_characteristics_2020]. Dyspnoea and pneumonia can subsequently develop within a median time of 8 days from disease onset [@wu_characteristics_2020]. Severe or critical cases (about 20% in some reports) require ventilation and intensive care unit (ICU) admission and can develop respiratory failure, septic shock and/or multiple organ dysfunction or failure [@wu_characteristics_2020]. Although being primarily a respiratory disease, multiple organs other than the lung and respiratory tract can be affected [@zaim_covid-19_2020], including the heart and blood vessels [@shi_association_2020; @bikdeli_covid-19_2020], liver [@zhang_liver_2020], kidneys [@diao_human_2020], brain [@mao_neurologic_2020] and intestine [@xiao_evidence_2020], among others. Although most patients recover from the disease, there can be long-term persisting symptoms (e.g. fatigue, shortness of breath, cough, joint pain and chest pain) [@cdc_covid-19_2020] and/or abnormalities found in clinical tests (e.g. pulmonary diffusion impairment) [@cortinovis_long-term_2021].

### SARS-CoV-2 biology

SARS-CoV-2 is a positive-sense single-stranded RNA (+ssRNA) enveloped virus from the *Coronaviridae* family, *Betacoronavirus* genus, and the *Sarbecovirus* subgenus [@zhu_novel_2020]. Its RNA genome has 29,891 bases, with a similar genomic organization with other betacoronaviruses [@dos_santos_natural_2020]. The genome encodes 29 proteins, including 6 functional open reading frames (ORFs) for the replicase (ORF1a/1b), spike (S), envelope (E), membrane (M) and nucleocapsid (N) proteins (the latter four being structural proteins), and 7 putative ORFs for accessory proteins [@chan_genomic_2020]. The replicase ORF encodes two large polyproteins that are proteolytically cleaved into 16 non-structural proteins (nsp1-16) [@chan_genomic_2020]. The spike protein is integrated into the viral envelope resulting in crown-like spikes on the surface of the virus, giving coronavirus its name [@dos_santos_natural_2020], which is also responsible for the viral attachment and entry of host cells via endocytosis [@shang_structural_2020; @walls_structure_2020]. It has been determined that the human angiotensin-converting enzyme 2 (ACE2) protein, present on the cell membrane, is the host receptor for viral entry [@yan_structural_2020]. The receptor binding motif (RBM) within the domain (RBD) of the viral spike protein trimer binds to the host ACE2 protein; host proteases including the serine protease TMPRSS2, cathepsin B/L and furin then cleave and "prime" the spike protein at the cell surface or in the early endosome [@hoffmann_sars-cov-2_2020; @shang_cell_2020], triggering a conformation change in the spike protein that subsequently promote membrane fusion and viral entry [@hoffmann_sars-cov-2_2020; @chambers_sars-cov-2_2020].

After entry into the host cell, the virus initiates a complicated process of protein expression and genome replication. The translation of the two replicase ORFs utilizes a programmed–1 ribosomal frameshift [@finkel_coding_2021], producing two polyproteins that will be cleaved into the 16 non-structural proteins (nsp1-16) via by two cysteine protease activities present in nsp3 and nsp5 [@vkovski_coronavirus_2020]. Nsp2-16 forms the virus replication and transcription complex (RTC) [@yan_architecture_2020], including the RNA-dependent RNA polymerase (RdRP, nsp12; with cofactors nsp7 and nsp8, as well as nsp14 which has an exonuclease domain for proofreading) for viral genome replication [@gao_structure_2020; @eckerle_high_2007]. In other coronaviruses, RTC is formed within double-membrane vesicles (DMVs) derived from endoplasmic reticulum (ER) and other host endomembranes; it is localized to specific subcellular locations involving complex interactions with host proteins to further advance the viral cell cycle [@vkovski_determination_2019]. The viral genome replication first produces a full-length negative-sense RNA, which is then further replicated into positive-sense genomic RNA [@vkovski_coronavirus_2020]. More viral proteins can then be translated from the newly synthesized viral genome. As with other coronaviruses, the genome replication is characterized by a unique discontinuous transcription at transcription regulatory sequences (TRSs), producing a nested set of subgenomic RNAs (sgRNAs) with shared 5' and 3' sequences [@sola_continuous_2015]. The virion assembly involves the association between the viral RNA genome with the nucleocapsid proteins and happens on the cytoplasmic side of the ER–Golgi intermediate compartment (ERGIC) cisternae [@klein_sars-cov-2_2020]. The assembled virions within membrane-bound vesicles are then released via exocytosis [@klein_sars-cov-2_2020; @ghosh_-coronaviruses_2020], completing the life cycle of the virus.

The host-pathogen interactions (HPI) of SARS-CoV-2 is highly complex. *In vitro* studies of SARS-CoV-2 protein expression followed by affinity purification and mass spectrometry (MS)-based protein identification has identified hundreds of host-SARS-CoV-2 protein-protein interactions (PPI) [@gordon_sars-cov-2_2020]. The host cell typically senses viral invasion and mounts anti-viral responses via different innate immune pathways, notably the type I/III interferon (IFN) pathway [@vabret_immunology_2020], however, SARS-CoV-2 is able to suppress the activity of these pathways via different mechanisms [@vabret_immunology_2020; @vkovski_coronavirus_2020]. For example, the ORF3b of the virus may effectively inhibit IFN response [@konno_sars-cov-2_2020], and it has been shown that SARS-CoV-2-infected cells and patients exhibit impaired IFN response [@blanco-melo_imbalanced_2020; @hadjadj_impaired_2020]. Other viral proteins, on the other hand, may induce the activation of various inflammatory pathways, e.g. interleukin (IL)-6 and IL-8 production that can cause the symptoms associated with severe diseases [@li_virus-host_2021]. It has been suggested that an imbalanced immune response involving the failure to mount an early IFN response and a dysregulated inflammatory response during SARS-CoV-2 infection drives the development of COVID-19 [@blanco-melo_imbalanced_2020; @giamarellos-bourboulis_complex_2020].

### Anti-COVID-19 therapies and vaccines

Governments, industries and scientific communities across the world and have responded quickly to the COVID-19 crisis in search of efficient preventative and therapeutic strategies. To date (March 2021), there are thousands of registered clinical trials on COVID-19 [@who_international_2021]. Potential therapies under clinical investigation include small-molecular drugs, biologics e.g. monoclonal antibodies, plasma-based and cell-based therapies, among others [@thorlund_real-time_2020; @noauthor_global_2020]. Unfortunately, Very few therapies have been established as beneficial to COVID-19 patients. Drugs proposed for COVID-19 treatment early in the pandemic including lopinavir, ritonavir and hydroxychloroquine were later shown to be ineffective [@recovery_collaborative_group_lopinavir-ritonavir_2020; @cavalcanti_hydroxychloroquine_2020]. Remdesivir, a viral RdRP inhibitor represents the only drug approved by the drug regulatory authorities of several countries including the U.S. Food and Drug Administration (FDA)  [@beigel_remdesivir_2020; @fda_fda_2020]. However, it confers only mild clinical benefits to a subset of COVID-19 patients, and is not recommended by the World Health Organization (WHO) as a first-line treatment for COVID-19 patients [@spinner_effect_2020; @who_solidarity_trial_consortium_repurposed_2020; @who_therapeutics_2020]. Janus kinase (JAK) inhibitor baricitinib (in combination with remdesivir), and virus-neutralizing antibodies bamlanivimab, and casirivimab plus imdevimab have obtained Emergency Use Authorization (EUA) from the FDA [@fda_coronavirus_2020; @fda_coronavirus_2020-1; @fda_coronavirus_2020-2]. Dexamethasone and other corticosteroids have been recommended by the U.S. National Institutes of Health (NIH) for hospitalized patients requiring supplemental oxygen [@recovery_collaborative_group_dexamethasone_2020; @nih_corticosteroids_2020]. To date, one of the most promising and encouraging advances among the anti-COVID-19 efforts has been the successful development of several SARS-CoV-2 vaccines around the world, some have been approved or authorized for emergency use in different countries [@dong_systematic_2020; @fda_pfizer-biontech_2021; @fda_moderna_2021].

There have been a lot of preclinical efforts in identifying potential therapeutic drugs and targets for anti-SARS-CoV-2. Numerous antiviral drug repurposing screens of varied scales using libraries of existent drugs have been performed *in vitro* [e.g. @riva_discovery_2020; @touret_vitro_2020]. Some studies have tried to identify effective drug combinations [e.g. @bobrowski_synergistic_2021; @nguyenla_discovery_2020]. A few genetic screens using e.g. the CRISPR-Cas9 technology have been reported, revealing potential gene targets for inhibiting SARS-CoV-2 infection or proliferation [@wei_genome-wide_2021; @daniloski_identification_2021; @wang_genetic_2021]. A recent study reported the application of CRISPR-Cas13 in degrading SARS-CoV-2 RNA and suppression the viral infection [@abbott_development_2020]. Different parties and research groups have created databases containing compiled resources on potential anti-SARS-CoV-2 drugs, targets and therapies [e.g. @kuleshov_covid-19_2020; @chen_litcovid_2021]. There are also large number of computation-driven efforts for predicting anti-SARS-CoV-2 drugs and/or targets, based on a wide variety of algorithms [e.g. @bobrowski_synergistic_2021; @zhou_network-based_2020; @zhou_artificial_2020]. 

### Viral regulation of host metabolism

Viruses are known to “hijack” host cell metabolism to complete their own intracellular life cycle [@mayer_hijacking_2019; @thaker_viral_2019], modulating diverse pathways including carbohydrate, lipid, amino acid and nucleotide metabolism [@mayer_hijacking_2019; @sanchez_viral_2015]. Coronaviruses including MERS-CoV rearrange cellular lipid profiles upon infection [@yan_characterization_2019; @yuan_srebp-dependent_2019]; as described above, coronaviruses induce the formation of complex networks of DMVs derived from ER and other intracellular membranes, in which they assemble their RTCs for viral replication and protein translation [@vkovski_determination_2019], and the formation of DMVs and RTCs are critical for the viral life cycle. Notably, counteracting the metabolic demands of viruses including MERS-CoV have been shown to abolish their ability to infect the host cells [@mayer_hijacking_2019; @yuan_srebp-dependent_2019]. Targeting the virus-induced metabolic changes has been proposed to be a promising novel antiviral strategy [@mayer_hijacking_2019].

Soon after the emergence of COVID-19, people have noticed that certain metabolic factors are important determinants of disease risk and severity [@marazuela_endocrine_2020]. Many reports have shown that diabetes patients have significantly increased risk of morbidity and mortality due to COVID-19 [e.g. @guan_clinical_2020; @petrilli_factors_2020]. Increase glucose level can promote SARS-COV-2 infection and proliferation in human monocytes [@codo_elevated_2020]. However, the full mechanisms underlying this association are yet unclear and likely involve systemic factors such as lung function and immune response being modulated by glucose level [@lim_covid-19_2021]. Obesity is another risk factor for hospitalization and severe disease of COVID-19 [@petrilli_factors_2020]. On the other hand, survivors of SARS may develop new-onset and long-lasting metabolic abnormalities, which may also occur in COVID-19 patients [@ayres_metabolic_2020]. Recent studies have reported that SARS-CoV-2 induces metabolic changes in numerous pathways including tricarboxylic acid (TCA) cycle, oxidative phosphorylation, lipid metabolism, amino acid metabolism (e.g. tryptophan metabolism), and pentose phosphate pathway, among others in human patient samples [@gardinassi_immune_2020; @ehrlich_sars-cov-2_2020; @thomas_covid-19_2020; @li_metabolic_2021; @bojkova_targeting_2020]. Several new studies have shown that anti-SARS-CoV-2 effect can be achieved by targeting various host metabolism pathways, for example, the PPARα-agonist fenofibrate can reverse SARS-CoV-2-induced metabolic changes and block the viral replication [@ehrlich_sars-cov-2_2020]; DHODH inhibitors that suppress pyrimidine *de novo* synthesis showed anti-SARS-CoV-2 activity [@xiong_novel_2020]; the inhibition of certain phosphoinositides can disrupt endocytosis and block SARS-CoV-2 cell-entry [@ou_characterization_2020]; and transketolase inhibitor was shown to inhibit SARS-CoV-2 in a dose-dependent manner [@bojkova_targeting_2020]. These studies suggest that targeting metabolism as an antiviral strategy can also be applied to SARS-CoV-2.

## Overview of study: rationale, aims and significance




