```{r, include=FALSE}
library(my.utils)
library(gembox)
library(ComplexHeatmap)
```


# The analysis and improvement of metabolic transformation algorithms {#mta}

## Introduction

Genome-scale metabolic modeling (GEM) algorithms have been applied to study cell metabolism under a wide variety of contexts, and have repeatedly proven to be valuable for generating accurate predictions and informative hypotheses [@gu_current_2019] (Section \@ref(intro-gem)). Here we focus on the metabolic transformation algorithm (MTA), initially developed in 2013 [@yizhak_model-based_2013], which is used for the prediction of metabolic targets whose KO can transform the cellular metabolic state from a given reference state to a given target state (Section \@ref(intro-gem-algo)). Apparently, this algorithm can be used for therapeutic target prediction for many metabolism-related diseases, by reversing the perturbed cell metabolism under the diseased state back to the normal healthy state. But thanks to its mechanism-based modeling approach under the GEM framework, MTA can also be more widely used to predict potential causal factors and perturbations beyond what can be identified from classical statistical inference. In the original study, MTA was used to predict lifespan-extending genes in the budding yeast *Saccharomyces cerevisiae*, and successfully identified two novel lifespan-extending metabolic genes (*adh2* and *gre3*) whose KO's were experimentally validated to significantly prolong the chronological lifespan of yeast [@yizhak_model-based_2013]. In a subsequent study, MTA was used to predict metabolic cancer driver genes, resulting in a novel cancer driver gene *FUT9* in colorectal cancer [@auslander_integrated_2017]. A more recent study applied MTA and successfully predicted mitochondrial dihydroorotate dehydrogenase (DHODH) involved in pyrimidine nucleotide biosynthesis as an effective therapeutic target for an intractable epilepsy called Dravet syndrome [@styr_mitochondrial_2019]. Despite repeated successful applications, the original algorithm was mostly validated in the bacteria *E. coli*, with only a few validations in higher organisms (2 validation datasets for mice and 2 for human); the performance in the mice and human datasets also appeared relatively weaker than those in the *E. coli* [@yizhak_model-based_2013]. Here I aim to perform a deeper investigation to comprehensively evaluate the performance of MTA in higher organisms especially human, also seeking to optimize the algorithm and further improve its prediction accuracy.

## Results


## Materials and Methods

### Softwares and code

R version 3.6.3 was used for all statistical tests. P values lesser than 2.22e-16 may not be computed accurately and are reported as “P<2.22e-16”. The Benjamini-Hochberg (BH) method was used for P value adjustment. The R packages ggplot2 (Wickham 2016) and ComplexHeatmap (Gu et al. 2016) were used to create the visualizations. Implementation of the new MTA algorithm can be found in the GitHub repository: https://github.com/ruppinlab/gembox.

## Discussion and Conclusion


