```{r, include=FALSE}
library(my.utils)
library(gembox)
library(ComplexHeatmap)
```


# Modeling of host metabolism during SARS-CoV-2 infection {#covid}

## Introduction

### COVID-19: natural history, epidemiology and clinical features 

The coronavirus disease 2019 (COVID-19) is a severe and highly infectious respiratory disease whose first known case appeared in late December 2019 in Wuhan, China [@zhu_novel_2020]. Since then, COVID-19 has quickly evolved into a global pandemic within a few months by March 2020 [@who_who_2020] and is still ongoing, resulting in more than 100 million infected cases and over 2 million deaths (as of March 2021) [@who_who_2020-1]. The pathogen for COVID-19 has been identified to be a new betacoronavirus that was never discovered before, and was named severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) [@coronaviridae_study_group_of_the_international_committee_on_taxonomy_of_viruses_species_2020]. Two related viruses of the coronavirus family (*Coronaviridae*), the severe acute respiratory syndrome coronavirus (SARS-CoV) and Middle East respiratory syndrome coronavirus (MERS-CoV), emerged in 2002 and 2012 respectively have also caused serious respiratory diseases [@cui_origin_2019], making COVID-19 the third coronavirus-related severe infectious disease discovered in the recent decades. Like SARS-CoV and MERS-CoV [@cui_origin_2019], SARS-CoV-2 is likely to be of zoonotic origin, with the first COVID-19 cases epidemiologically linked to a wet market in Wuhan selling seafood, live poutry and wildlifes [@jiang_emerging_2020]. Phylogenetic studies have identified high similarities between the genomic sequences of SARS-CoV-2 and a group of betacoronaviruses of the subgenus *Sarbecovirus* previously found in bats in China [@wu_new_2020; @zhou_pneumonia_2020], and SARS-CoV-2-related coronaviruses have aslo been found in Malayan pangolins (*Manis javanica*) [@lam_identifying_2020]. However, the identity of the direct progenitor of SARS-CoV-2 is still uncertain [@andersen_proximal_2020], and the intermediate host responsible for the zoonotic transmission of the virus from the wild reservoir host species to human has not been confidently identified [@zhao_potential_2020].

Airborne transmission has been determined to be the dominant route of transmission of COVID-19 [@zhang_identifying_2020], specifically via respiratory droplets and potentially aerosol [@van_doremalen_aerosol_2020], fecal-oral transmission can also be possible [@xu_characteristics_2020]. COVID-19 is highly transmissible, with a basic reproduction number (the expected number of infections directly caused by transmission from one primary infected individual when all individuals are susceptible) estimated to be around 2 or 3 [@zhao_preliminary_2020; @billah_reproductive_2020], similar to SARS and higher than e.g. the 2009 influenza and MERS [@petersen_comparing_2020]. The incubation period (time from infection to symptom onset) of COVID-19 was estimated to range from 1.3 to 11.3 days [@lauer_incubation_2020], longer than SARS [@petersen_comparing_2020]. SARS-CoV-2 show a distinct kinetics of virus shedding, with high viral load when symptoms begin and a subsequent decline but persisting for a long period, making it critical to timely isolate and quarantine patients as soon as infection is detected and for sufficiently long time [@wolfel_virological_2020; @zheng_viral_2020; @petersen_comparing_2020]. COVID-19 is less deadly than SARS or MERS, with possibly more than 50% infected cases being asymptomatic [@li_asymptomatic_2020], and most symptomatic cases being mild [@wu_characteristics_2020]. However, asymptomatic patients can still spread the virus [@li_asymptomatic_2020], adding to the challenge of transmission control. The infection-fatality risk was estimated to be 1.39% in the New York City, the first epidemic centre in the USA [@yang_estimating_2021], and other estimates for other regions in the world were mostly below 1% [@streeck_infection_2020; @pastor-barriuso_infection_2020]. Older age and various comorbidities including hypertension, obesity, diabetes and coronary heart disease (among others) were identified as the risk factors for SARS-CoV-2 infection and/or severe disease [@mao_assessing_2020; @jordan_who_2020].

The most common initial symptoms of COVID-19 include dry cough, fever, and fatigue, additional symptoms can include sputum production, sore throat, haemoptysis, chills, headache, chest pain, anorexia, diarrhoea, and nausea/vomiting [@hu_characteristics_2020]. Dyspnoea and pneumonia can subsequently develop within a median time of 8 days from disease onset [@wu_characteristics_2020]. Severe or critical cases (about 20% in some reports) require ventilation and intensive care unit (ICU) admission and can develop respiratory failure, septic shock and/or multiple organ dysfunction or failure [@wu_characteristics_2020]. Although being primarily a respiratory disease, multiple organs other than the lung and respiratory tract can be affected [@zaim_covid-19_2020], including the heart and blood vessels [@shi_association_2020; @bikdeli_covid-19_2020], liver [@zhang_liver_2020], kidneys [@diao_human_2020], brain [@mao_neurologic_2020] and intestine [@xiao_evidence_2020], among others. Although most patients recover from the disease, there can be long-term persisting symptoms (e.g. fatigue, shortness of breath, cough, joint pain and chest pain) [@cdc_covid-19_2020] and/or abnormalities found in clinical tests (e.g. pulmonary diffusion impairment) [@cortinovis_long-term_2021].

### SARS-CoV-2 biology

SARS-CoV-2 is a positive-sense single-stranded RNA (+ssRNA) enveloped virus from the *Coronaviridae* family, *Betacoronavirus* genus, and the *Sarbecovirus* subgenus [@zhu_novel_2020]. Its RNA genome has 29,891 bases, with a similar genomic organization with other betacoronaviruses [@dos_santos_natural_2020]. The genome encodes 29 proteins, including 6 functional open reading frames (ORFs) for the replicase (ORF1a/1b), spike (S), envelope (E), membrane (M) and nucleocapsid (N) proteins (the latter four being structural proteins), and 7 putative ORFs for accessory proteins [@chan_genomic_2020]. The replicase ORF encodes two large polyproteins that are proteolytically cleaved into 16 non-structural proteins (nsp1-16) [@chan_genomic_2020]. The spike protein is integrated into the viral envelope resulting in crown-like spikes on the surface of the virus, giving coronavirus its name [@dos_santos_natural_2020], which is also responsible for the viral attachment and entry of host cells via endocytosis [@shang_structural_2020; @walls_structure_2020]. It has been determined that the human angiotensin-converting enzyme 2 (ACE2) protein, present on the cell membrane, is the host receptor for viral entry [@yan_structural_2020]. The receptor binding motif (RBM) within the domain (RBD) of the viral spike protein trimer binds to the host ACE2 protein; host proteases including the serine protease TMPRSS2, cathepsin B/L and furin then cleave and "prime" the spike protein at the cell surface or in the early endosome [@hoffmann_sars-cov-2_2020; @shang_cell_2020], triggering a conformation change in the spike protein that subsequently promote membrane fusion and viral entry [@hoffmann_sars-cov-2_2020; @chambers_sars-cov-2_2020].

After entry into the host cell, the virus initiates a complicated process of protein expression and genome replication. The translation of the two replicase ORFs utilizes a programmed–1 ribosomal frameshift [@finkel_coding_2021], producing two polyproteins that will be cleaved into the 16 non-structural proteins (nsp1-16) via by two cysteine protease activities present in nsp3 and nsp5 [@vkovski_coronavirus_2020]. Nsp2-16 forms the virus replication and transcription complex (RTC) [@yan_architecture_2020], including the RNA-dependent RNA polymerase (RdRP, nsp12; with cofactors nsp7 and nsp8, as well as nsp14 which has an exonuclease domain for proofreading) for viral genome replication [@gao_structure_2020; @eckerle_high_2007]. In other coronaviruses, RTC is formed within double-membrane vesicles (DMVs) derived from endoplasmic reticulum (ER) and other host endomembranes; it is localized to specific subcellular locations involving complex interactions with host proteins to further advance the viral cell cycle [@vkovski_determination_2019]. The viral genome replication first produces a full-length negative-sense RNA, which is then further replicated into positive-sense genomic RNA [@vkovski_coronavirus_2020]. More viral proteins can then be translated from the newly synthesized viral genome. As with other coronaviruses, the genome replication is characterized by a unique discontinuous transcription at transcription regulatory sequences (TRSs), producing a nested set of subgenomic RNAs (sgRNAs) with shared 5' and 3' sequences [@sola_continuous_2015]. The virion assembly involves the association between the viral RNA genome with the nucleocapsid proteins and happens on the cytoplasmic side of the ER–Golgi intermediate compartment (ERGIC) cisternae [@klein_sars-cov-2_2020]. The assembled virions within membrane-bound vesicles are then released via exocytosis [@klein_sars-cov-2_2020; @ghosh_-coronaviruses_2020], completing the life cycle of the virus.

The host-pathogen interactions (HPI) of SARS-CoV-2 is highly complex. *In vitro* studies of SARS-CoV-2 protein expression followed by affinity purification and mass spectrometry (MS)-based protein identification has identified hundreds of host-SARS-CoV-2 protein-protein interactions (PPI) [@gordon_sars-cov-2_2020]. The host cell typically senses viral invasion and mounts anti-viral responses via different innate immune pathways, notably the type I/III interferon (IFN) pathway [@vabret_immunology_2020], however, SARS-CoV-2 is able to suppress the activity of these pathways via different mechanisms [@vabret_immunology_2020; @vkovski_coronavirus_2020]. For example, the ORF3b of the virus may effectively inhibit IFN response [@konno_sars-cov-2_2020], and it has been shown that SARS-CoV-2-infected cells and patients exhibit impaired IFN response [@blanco-melo_imbalanced_2020; @hadjadj_impaired_2020]. Other viral proteins, on the other hand, may induce the activation of various inflammatory pathways, e.g. interleukin (IL)-6 and IL-8 production that can cause the symptoms associated with severe diseases [@li_virus-host_2021]. It has been suggested that an imbalanced immune response involving the failure to mount an early IFN response and a dysregulated inflammatory response during SARS-CoV-2 infection drives the development of COVID-19 [@blanco-melo_imbalanced_2020; @giamarellos-bourboulis_complex_2020].

### Anti-COVID-19 therapies and vaccines

Governments, industries and scientific communities across the world and have responded quickly to the COVID-19 crisis in search of efficient preventative and therapeutic strategies. To date (March 2021), there are thousands of registered clinical trials on COVID-19 [@who_international_2021]. Potential therapies under clinical investigation include small-molecular drugs, biologics e.g. monoclonal antibodies, plasma-based and cell-based therapies, among others [@thorlund_real-time_2020; @noauthor_global_2020]. Unfortunately, Very few therapies have been established as benefitial to COVID-19 patients. Drugs proposed for COVID-19 treatment early in the pandemic including lopinavir, ritonavir and hydroxychloroquine were later shown to be ineffective [@recovery_collaborative_group_lopinavir-ritonavir_2020; @cavalcanti_hydroxychloroquine_2020]. Remdesivir, a viral RdRP inhibitor represents the only drug approved by the drug regulatory authorities of several countries including the U.S. Food and Drug Administration (FDA)  [@beigel_remdesivir_2020; @fda_fda_2020]. However, it confers only mild clinical benefits to a subset of COVID-19 patients, and is not recommended by the World Health Organization (WHO) as a first-line treatment for COVID-19 patients [@spinner_effect_2020; @who_solidarity_trial_consortium_repurposed_2020; @who_therapeutics_2020]. Janus kinase (JAK) inhibitor baricitinib (in combination with remdesivir), and virus-neutralizing antibodies bamlanivimab, and casirivimab plus imdevimab have obtained Emergency Use Authorization (EUA) from the FDA [@fda_coronavirus_2020; @fda_coronavirus_2020-1; @fda_coronavirus_2020-2]. Dexamethasone and other corticosteroids have been recommended by the U.S. National Institutes of Health (NIH) for hospitalized patients requiring supplemental oxygen [@recovery_collaborative_group_dexamethasone_2020; @nih_corticosteroids_2020]. To date, one of the most promising and encouraging advances among the anti-COVID-19 efforts has been the successful development of several SARS-CoV-2 vaccines around the world, some have been approved or authorized for emergency use in different countries [@dong_systematic_2020; @fda_pfizer-biontech_2021; @fda_moderna_2021].

There have been a lot of preclinical efforts in identifying potential therapeutic drugs and targets for anti-SARS-CoV-2. Numerous antiviral drug repurposing screens of varied scales using libraries of existent drugs have been performed *in vitro* [e.g. @riva_discovery_2020; @touret_vitro_2020]. Some studies have tried to indentify effective drug combinations [e.g. @bobrowski_synergistic_2021; @nguyenla_discovery_2020]. A few genetic screens using e.g. the CRISPR-Cas9 technology have been reported, revealing potential gene targets for inhibiting SARS-CoV-2 infection or proliferation [@wei_genome-wide_2021; @daniloski_identification_2021; @wang_genetic_2021]. A recent study reported the application of CRISPR-Cas13 in degrading SARS-CoV-2 RNA and suppression the viral infection [@abbott_development_2020]. Different parties and research groups have created databases containing compiled resources on potential anti-SARS-CoV-2 drugs, targets and therapies [e.g. @kuleshov_covid-19_2020; @chen_litcovid_2021]. There are also large number of computation-driven efforts for predicting anti-SARS-CoV-2 drugs and/or targets, based on a wide variety of algorithms [e.g. @bobrowski_synergistic_2021; @zhou_network-based_2020; @zhou_artificial_2020]. 

### Viral regulation of host metabolism

Viruses are known to “hijack” host cell metabolism to complete their own intracellular life cycle [@mayer_hijacking_2019; @thaker_viral_2019], modulating diverse pathways including carbohydrate, lipid, amino acid and nucleotide metabolism [@mayer_hijacking_2019; @sanchez_viral_2015]. Coronaviruses including MERS-CoV rearrange cellular lipid profiles upon infection [@yan_characterization_2019; @yuan_srebp-dependent_2019]; as described above, coronaviruses induce the formation of complex networks of DMVs derived from ER and other intracellular membranes, in which they assemble their RTCs for viral replication and protein translation [@vkovski_determination_2019], and the formation of DMVs and RTCs are critical for the viral life cycle. Notably, counteracting the metabolic demands of viruses including MERS-CoV have been shown to abolish their ability to infect the host cells [@mayer_hijacking_2019; @yuan_srebp-dependent_2019]. Targeting the virus-induced metabolic changes has been proposed to be a promising novel antiviral strategy [@mayer_hijacking_2019].

Soon after the emergence of COVID-19, people have noticed that certain metabolic factors are important determinants of disease risk and severity [@marazuela_endocrine_2020]. Many reports have shown that diabetes patients have significantly increased risk of morbidity and mortality due to COVID-19 [e.g. @guan_clinical_2020; @petrilli_factors_2020]. Increase glucose level can promote SARS-COV-2 infection and proliferation in human monocytes [@codo_elevated_2020]. However, the full mechanisms underlying this association are yet unclear and likely involve systemic factors such as lung function and immune response being modulated by glucose level [@lim_covid-19_2021]. Obesity is another risk factor for hospitalization and severe disease of COVID-19 [@petrilli_factors_2020]. On the other hand, survivors of SARS may develop new-onset and long-lasting metabolic abnormalities, which may also occur in COVID-19 patients [@ayres_metabolic_2020]. Recent studies have reported that SARS-CoV-2 induces metabolic changes in numerous pathways including tricarboxylic acid (TCA) cycle, oxidative phosphorylation, lipid metabolism, animo acid metabolism (e.g. tryptophan metabolism), and pentose phosphate pathway, among others in human patient samples [@gardinassi_immune_2020; @ehrlich_sars-cov-2_2020; @thomas_covid-19_2020; @li_metabolic_2021; @bojkova_targeting_2020]. Several new studies have shown that anti-SARS-CoV-2 effect can be achieved by targeting various host metabolism pathways, for example, the PPARα-agonist fenofibrate can reverse SARS-CoV-2-induced metabolic changes and block the viral replication [@ehrlich_sars-cov-2_2020]; DHODH inhibitors that suppress pyrimidine *de novo* synthesis showed anti-SARS-CoV-2 activity [@xiong_novel_2020]; the inhibition of certain phosphoinositides can disrupt endocytosis and block SARS-CoV-2 cell-entry [@ou_characterization_2020]; and transketolase inhibitor was shown to inhibit SARS-CoV-2 in a dose-dependent manner [@bojkova_targeting_2020]. These studies suggest that targeting metabolism as an antiviral strategy can also be applied to SARS-CoV-2.

### Rationale, aims and approaches of study

Despite unprecedented global efforts to control the COVID-19 pandemic, this global health crisis has not been resolved to date (as of March 2021). While the wide deployment of vaccination is expected to significantly slow down the spread of the disease, the evolution and emergence of variant virus strains suggests a possible need to update the vaccines in the future [@wang_mrna_2021]. Some studies predicted that there may be recurrent wintertime COVID-19 outbreaks over a longer term after the initial pandemic [@kissler_projecting_2020]. Currently, effective therapeutic options are still very scarce, and thus the development of new treatment strategies are still urgently needed for long-term control of the pandemic. Drug repurposing and combination may represent practical and time-efficient strategies to address this urgent unmet medical need. Given the recognized importance of host metabolism in coronavirus (including SARS-CoV-2) infection and proliferation, we aim to identify anti-SARS-CoV-2 targets that act via counteracting the virus-induced metabolic changes, either individually or in combination with current anti-SARS-CoV-2 drugs, esp. remdesivir. Such a metabolic-targeting approach may be especially valuable for anti-SARS-CoV-2 drug repurposing to address the current urgent COVID-19 crisis considering that many existing drugs are metabolism-targeting.

Our specific approach involves the application of genome-scale metabolic modeling (GEM) in an integrated analysis of multiple published *in vitro* and human patient gene expression datasets on SARS-CoV-2 infection, in order to comprehensively identify SARS-CoV-2 induced host metabolic reprogramming. Then we applied our improved version of the metabolic transformation algorithm (MTA) to predict the metabolic anti-SARS-CoV-2 targets. The predictions were validated with the experimentally identified anti-SARS-CoV-2 drugs and targets reported in published drug or genetic screens.

## Results

### Analysis of SARS-CoV-2-induced gene expression changes across datasets

### xxx

## Materials and Methods

text

## Conclusion and Discussion

text
  
