```{r, include=FALSE}
library(my.utils)
library(gembox)
library(ComplexHeatmap)
```


# Metabolic modeling of CD8 T cell for anti-cancer immunotherapy ^[The work of this chapter is a collaborative project with Dr. Madhusudhanan Sukumar in Dr. Nicholas Restifo's lab at NCI, who conducted the biological experiments investigating the findings based on computational predictions. All biological experiment results included are credited to Dr. Madhusudhanan Sukumar, after obtaining the permission of him and Dr. Nicholas Restifo. The contents of this chapter have been written into a manuscript on which I and Dr. Madhusudhanan Sukumar are co-first authors and will be submitted to a journal. Many contents of this chapter are taken from the above manuscript, involving direct quotes with minor rephrasing.] {#ucp2}

## Introduction

Various cancer immunotherapies, notably immune checkpoint blockade (ICB) and adoptive cell transfer (ACT) therapies have revolutionized the landscape of cancer therapeutics, achieving unprecedented durable response. However, various challenges still remain, including the increasing the currently suboptimal response rate in ICB therapy and enhancing the long-term persistence of transfused T cells in ACT therapies (Sections \@ref(intro-icb) and \@ref(intro-act)). T cells can play a pivotal role in anti-cancer immunity, and their functional state is an important determinant of response to current cancer immunotherapies. It is therefore desirable to further elucidate the factors regulating T cell function in order to improve the success of the immunotherapies. It has been established that metabolism can greatly affect different aspects of T cell function, including cytolytic activity, differentiation, longevity and exhaustion (Section \@ref(intro-t-metab)), with mitochondrial metabolism lying at the center of T cell metabolic regulation [@desdin-mico_mitochondrial_2018]. Modulating T cell mitochondrial metabolism has further been proposed to represent a promising strategy to boost immunotherapy response [@li_rewiring_2020]. Here I aim to take advantage of the framework of genome-scale metabolic modeling (GEM) integrated with publicly available T cell transcriptomic data in cancer immunotherapy to understand the global metabolic features associated with T cell function and immunotherapy response. I specifically applied the metabolic transformation algorithm (MTA; Section \@ref(intro-gem-algo) and Chapter \@ref(mta)) on various ACT datasets to predict metabolic processes that are essential for T cell function in the ACT setting. Combining with the analysis of additional ICB datasets and the cancer patient data from the TCGA project, I identified mitochondrial uncoupling and specifically the UCP2 protein as important for therapy response. Via collaboration with experimentalists, the role and mechanisms of UCP2 in regulating T cell function and immunotherapy response was further comprehensively investigated in the mice B16 melanoma model. The extensive biological experiments confirmed that UCP2 is required for T cell stemness and affects its anti-tumor function in mice, and also further elucidated an ROS and mTOR signaling-related mechanisms, echoing previous studies that established the important roles of ROS and mTOR [@klein_geltink_unraveling_2018; @powell_regulation_2012; Section \@ref(intro-t-metab)].

## Results

### GEM-based analysis of T cell metabolism in cancer immunotherapy

df analysis

### MTA identified T cell mitochondrial uncoupling as essential for the anti-cancer function and immunotherapy response

To identify novel candidate metabolic regulators that are important for the T cell anti-tumor function, we start with a computational approach using genome-scale metabolic modeling (GEM). Specifically, we applied the metabolic transformation algorithm (MTA) (Yizhak et al., 2013) on a published gene expression dataset of CAR-T cells for anti-CD19 CAR-T therapy in chronic lymphocytic leukemia (CLL) (Fraietta et al., 2018), and predicted metabolic reactions whose knockout (KO) can transform the responder-associated T cell metabolic state to that of the non-responders. Briefly, MTA computes sample-specific metabolic profile as the rates (i.e. fluxes) of all cellular metabolic reactions based on the gene expression, then simulates the KO of each reaction with GEM under the T cell metabolic background of the responders; as outputs, MTA ranks all metabolic reaction KO’s by their capacity in leading to a closer metabolic state to that of the non-responders (illustrated in Figure 1A; Methods). Therefore, top MTA-predicted metabolic reactions represent candidates that are essential for potent T cell anti-tumor function, since their KO’s are predicted to result in non-responsiveness in CAR-T therapy. We performed pathway enrichment analysis of the top 10% of MTA-predicted reactions (Figure 1B), and found that MTA was able to successfully recover known metabolic pathways that regulate T cell function, for example those related to nucleotide and amino acid metabolism (ref), and in particular oxidative phosphorylation (OXPHOS) (ref). OXPHOS has been shown to … (ref). Consistently we found that the electron transfer chain (ETC) and ATP synthase fluxes are overall higher in the CAR-T cells from the responders vs that in the non-responders, as computed via GEM using the same anti-CD19 CAR-T dataset (Figure 1C; Methods). Using another dataset on KRAS(G12D)-targeting tumor-infiltrating lymphocytes (TILs) for adoptive cell transfer therapy (ACT) in a metastatic colorectal cancer patient (Lu et al., 2019), we again confirmed overall higher OXPHOS-related metabolic fluxes computed via GEM in a T cell clone that persisted for up to 266 days after transfer ("persistent clone 9.2-P") vs another clone targeting the same KRAS(G12D) epitope but did not persist for longer than 40 days ("non-persistent clone 9.1-NP"; Figure 1D; Methods). These results showcase that GEM and in particular the MTA algorithm is valuable in identifying metabolic determinants of T cell function and longevity.

Next, we examined the top predictions made by MTA to identify candidates of novel metabolic regulators of T cell function. The top metabolic pathway enriched by MTA predictions is "Transport, Mitochondrial" (hypergeometric test FDR<0.05, Figure 1B). We therefore focused on this pathway and further separated all mitochondrial transport reactions by the metabolite being transported; while the transport of various nucleotides are enriched, we found that the top enriched metabolite is proton (H+, hypergeometric test FDR<0.1, Figure 1E). We note that these proton transport reactions include the various co-/anti-transporters of proton with other metabolites, but do not include the ETC and ATP synthase reactions. The enrichment of mitochondrial transport as well as proton transport specifically was also confirmed with the GSEA method (Figure 1F), suggesting that mitochondrial proton transport is critical for T cell function.
	
Since proton can be transported both into and out of the mitochondria, we aimed to determine whether T cell function is associated with a particular direction of proton transport. We found that the top MTA predictions mostly include the subset of proton transport reactions contributing to a higher mitochondrial proton influx in the anti-CD19 CAR-T responders vs non-responders (Wilcoxon rank-sum test P=0.0x, Figure 1G). Together with the observation that the responders also have higher total mitochondrial proton influx (Figure 1H), these suggest that mitochondrial proton inflow, rather than outflow, is specifically important for the T cell anti-tumor function. We further examined the mitochondrial uncoupling reaction, the key reaction responsible for proton release into the mitochondrial matrix, and found that consistently this reaction has higher flux in the responders vs non-responders (Figure 1I), and also in the persistent vs non-persistent T cell clone (Figure 1J) from the anti-KRAS(G12D) TIL dataset as described above (Lu et al., 2019). Together, these results point to mitochondrial uncoupling as an important regulator of T cell function and longevity.

### Analysis of large-scale human transcriptome datasets identified UCP2 as the major gene for mitochondiral uncoupling in immune cells

There are three major mitochondrial uncoupling proteins (UCPs), namely UCP1, UCP2, and UCP3, located within the inner mitochondrial membrane with close proximity to ETC proteins (Krauss et al., 2005). Little is known about their function in CD8+ T cells. Analyzing human tissue gene expression data from the Human Protein Atlas (ref). We see that among the three genes UCP2 has much higher expression levels than UCP1 and UCP3 across tissues, and shows especially abundant levels in the PBMC, thymus and spleen, suggesting immune cell-associated expression (Figure 1K), which is consistent with a previous report (Rupprecht et al., 2012). Furthermore, we performed a correlation analysis in each of the cancer types available from the TCGA dataset (ref); strikingly we found a very prevalent positive correlation between the expression levels of UCP2 and a few T cell memory or stemness marker genes (including CD8A, TCF7, Il7R and SELL) across cancer types, after controlling for tumor purity (Figure 1L; Methods). Such correlations are not found for UCP1 or UCP3 (Figure S1x). Higher UCP2 expression level is also associated with better overall survival in melanoma patients from TCGA (Figure S1x). These results further support the role of UCP2 in CD8+ T cell anti-tumor function, which we further investigated in depth.

### Ucp2 inhibition in mice T cells decreased T cell stemness marker expression and reduced anti-cancer function

### GEM modeling supports increased reactive oxygen species as a mechanism underlying the effect of Ucp2

### Additional experiments in mice melanoma model validated the ROS-related mechanisms for the effect of Ucp2

### MTA prediction of metabolic targets for improving T cell anti-cancer function

## Materials and Methods

### Differential expression and gene set enrichment analysis

Differential expression (DE) analysis of responders vs non-responders of various cancer immunotherapy datasets, as well as for the mice T cells between different experimental conditions (control, NAC and/or genipin treatment) were performed with edgeR [@robinson_edger_2010]. For gene set enrichment analysis (GSEA) [@subramanian_gene_2005] from the DE log fold-change results, an implementation of the GSEA algorithm in the R package fgsea [@korotkevich_fast_2021] was used, with gene set annotations were taken from the MSigDB database [@liberzon_molecular_2011]. Specifically, the "C7 immunologic signature gene sets" from MSigDB v7.0 was used to analyze the differences in T cell activation or memory functions across the mice T cell samples treated by NAC and/or genipin.

### Genome-scale metabolic modeling-based prediction of metabolic functions determining response to anti-CD19 CAR-T therapy

Gene expression data of the CAR-T cells for responders and non-responders was obtained from Fraietta et al. [ref]. The median expression profiles across all responders and all non-responders were computed respectively, then the iMAT algorithm [ref] was used to obtain the condition-specific metabolic models of the two groups (responder and non-responder), and then ACHR sampling [ref] was used to obtain the fluxes distributions of all metabolic reactions. A variant of the MTA algorithm [ref], called rMTA [refs] was then applied to predict metabolic reactions whose knock-out can transform cellular metabolic state from that of the responder to that of the non-responder. The output of rMTA is a score (rMTA score) for each metabolic reaction, with higher scores corresponding to better candidates for achieving the metabolic transformation as specified above. For the genome-scale metabolic modeling, the human Recon 1 model was used [ref].

### Metabolic pathway analysis of the MTA-predicted metabolic reactions

The "subSystems" data slot from the human Recon 1 genome-scale metabolic model was used as metabolic pathway annotations for each metabolic reaction (i.e. each "sub system" is treated as one metabolic pathway). Fisher’s exact test was used to test for significant enrichment of each metabolic pathway in the top 10% reactions predicted by rMTA as described above. P values were corrected for multiple hypotheses with the Benjamini-Hochberg method. Further, to closely analyze the enrichment for mitochondrial transport reactions in a metabolite-specific manner, the transport reactions across the mitochondrial membrane for each metabolite were identified from the Recon 1 model (the model does not distinguish between the inner and outer membrane of the mitochondria), and these were used as reaction sets. The enrichment of top 10% rMTA-predicted reactions in these reaction sets were tested with Fisher’s exact test similarly as above. As an independent approach, the GSEA method [ref] was also used to test for the enrichment of top rMTA-predicted reactions in the "Transport, Mitochondrial" pathway and specifically the set of mitochondrial proton transport reactions. For this the pathway/reaction set was defined as above, and all metabolic reactions ranked by their rMTA score was passed to the GSEA algorithm to compute the result.

### Genome-scale metabolic modeling-based differential flux analysis

Genome-scale metabolic modeling was used to compute the metabolic flux distributions from the gene expressions profiles of different samples analyzed or generated in our study. These include the CAR-T cells from responders and non-responders in anti-CD19 CAR-T therapy (Fraietta et al.), the persistent and non-persistent T cell clones in an adoptive T cell transfer therapy (Lu et al.), and the mice T cell samples treated by NAC and/or genipin generated in our study. For human samples/data, the Recon 1 model was used, and for mice samples, the mouse genome-scale metabolic model (iMM1415) [ref] was used for the modeling and was obtained from the BiGG database [ref]. The expression profiles were averaged across samples in each group or experimental condition by taking the median, and iMAT [ref] was applied to the median expression profiles, generating group/condition-specific metabolic models. ACHR sampling [ref] was then used to generate the fluxes distributions of all metabolic reactions. We applied iMAT to model the median expression in each group or experimental condition to reduce data noise; this is also due to that iMAT is not efficient in modeling metabolic flux from the similar samples within each group. The resulting flux distributions are smooth (rather than discrete with replicates like experimentally collected data), and are meant to be the representative flux distribution for each group/condition; the variation in the distribution reflects uncertainty in the computation and flexibility in the metabolic space, rather than variation across samples. Therefore, subsequent comparisons of the flux distributions between two groups/conditions were carried out in a qualitative manner by plotting and visually inspecting the smooth flux distributions and not by statistical tests (as these distributions are equivalent to population distributions rather than sample distributions).

### Applying metabolic modeling to identify the major targets of genipin

Based on the RNA-seq data of the vehicle and genipin-treated CD8+ T cells and the mouse metabolic model iMM1415 as described above, the metabolic transformation algorithm (MTA)[ref] was used to identify the metabolic gene(s) whose inhibition can transform the cellular metabolic state from that of the vehicle cells to that of the genipin-treated cells, thus effectively identifying the targets of genipin. MTA ranks all the metabolic reactions based on their probability of being the target of genipin, and UCP2-catalyzed reaction was ranked among the top 2% of all reactions, strongly supporting that UCP2 is a major target of genipin.

### Softwares and code

R version 3.6.3 was used for all statistical tests. P values lesser than 2.22e-16 may not be computed accurately and are reported as "P<2.22e-16". The Benjamini-Hochberg (BH) method was used for P value adjustment. The R packages ggplot2 [@wickham_ggplot2_2009] was used to create the visualizations. Code for the analyses can be found in the GitHub repository: https://github.com/ruppinlab/Tcell_Ucp2.

## Discussion and Conclusion


