```{r, include=FALSE}
library(my.utils)
library(gembox)
library(ComplexHeatmap)
```


# Modeling of host metabolism during SARS-CoV-2 infection {#covid} ^[The work of this chapter has been organized in the journal article format and submitted to the online preprint server bioRxiv (@cheng_genome-scale_2021); it has also been submitted to a journal, which is under peer-review as of Mar 2020. Many contents of this chapter are taken from the above article, involving direct quotes with minor rephrasing.]

## Introduction

Despite unprecedented global efforts to control the COVID-19 pandemic, this global health crisis has not been resolved to date (as of March 2021). While the wide deployment of vaccination is expected to significantly slow down the spread of the disease, the evolution and emergence of variant virus strains suggests a possible need to update the vaccines in the future [@wang_mrna_2021]. Some studies predicted that there may be recurrent wintertime COVID-19 outbreaks over a longer term after the initial pandemic [@kissler_projecting_2020]. Currently, effective therapeutic options are still very scarce, and thus the development of new treatment strategies are still urgently needed for long-term control of the pandemic. Drug repurposing and combination may represent practical and time-efficient strategies to address this urgent unmet medical need. Given the recognized importance of host metabolism in coronavirus (including SARS-CoV-2) infection and proliferation, we aim to identify anti-SARS-CoV-2 targets that act via counteracting the virus-induced metabolic changes, either individually or in combination with current anti-SARS-CoV-2 drugs, esp. remdesivir. Such a metabolic-targeting approach may be especially valuable for anti-SARS-CoV-2 drug repurposing to address the current urgent COVID-19 crisis considering that many existing drugs are metabolism-targeting.

My specific approach involves the application of genome-scale metabolic modeling (GEM) in an integrated analysis of multiple published *in vitro* and human patient gene expression datasets on SARS-CoV-2 infection, in order to comprehensively identify SARS-CoV-2 induced host metabolic reprogramming. Then I applied my improved version of the metabolic transformation algorithm (MTA) to predict the metabolic anti-SARS-CoV-2 targets. The predictions were validated with the experimentally identified anti-SARS-CoV-2 drugs and targets reported in published drug or genetic screens.

## Results

### Analysis of SARS-CoV-2-induced gene expression changes across datasets

Multiple studies have characterized the gene expression changes during SARS-CoV-2 infection in different *in vitro* and *in vivo* settings. A systematic meta-analysis of published datasets can be desirable in order to identify and summarize robust and recurrent virus-induced molecular changes. We collected a total of 12 published relevant datasets spanning a wide range of sample types (various cell lines, primary bronchial epithelial cells, nasopharyngeal swab and bronchoalveolar lavage fluid, i.e. BALF samples from patients) and assay platforms (bulk RNA-seq, scRNA-seq, and MS-based proteomics). These datasets are summarized in Table 1. With each of the datasets, we performed differential expression (DE) analysis comparing the SARS-CoV-2-infected or positive samples to the non-infected control or negative samples. For the single-cell datasets, we focused only on the airway epithelial cell which is known as the major virus-infected cell type. Comparing the datasets with a principle component analysis (PCA) plot based on the inverse normal-transformed DE log fold-change values (Fig. 1A; Methods) suggests that the cell lines tend to have distinct DE profiles from the patient samples, although different patient datasets exhibit considerable variation depending on sample type and sequencing platform. This pattern is confirmed by the comparison of the top significant DE genes from each pair of datasets (using Fisher’s exact tests, top n=400 DE genes uniformly from each datasets were used, all with at least FDR<0.1; Methods; Fig. 1B). Moreover, examining only the top DE genes appears to mitigate the technical variation across datasets, with reasonable coherence demonstrated by odds ratio median value 2.18 and maximum 13.60 (adjusted P median 5.09e-5, minimum<2.22e-16; Fig. 1B).

We then performed gene set enrichment analysis (GSEA) (Subramanian et al. 2005) on the DE results from each dataset (Table S2), and further compared the datasets on the pathway level by the significantly enriched pathways (FDR<0.1; Methods). Reassuringly, the level of coherence across datasets on the pathway level is even stronger, with a median odds ratio of 4.25 (maximum is infinity followed by 35.49) across pairs of datasets (adjusted P median 2.01e-5, minimum<2.22e-16; Fig. 1C). Although no pathway is consistently significant across all 12 datasets, examining the most consistently enriched pathways across the datasets while giving higher importance to the various in vivo patient datasets (Fig. 1D; Table S3; Methods), we see many up-regulated pathways involved in innate immune response to viral infection, e.g. interferon signaling and cytosolic DNA sensing. Among the pathways involving coherently down-regulated genes upon SARS-CoV-2 infection, we find antigen presentation, as well as numerous pathways spanning many major categories of cellular metabolism, e.g. oxidative phosphorylation (OXPHOS; energy metabolism), sphingolipid metabolism (lipid metabolism), valine, leucine and isoleucine degradation (amino acid metabolism), and N-glycan biosynthesis, among others, which may reflect the specific metabolic requirements of SARS-CoV-2 or underlie its pathogenic effects (see Discussion). Visualizing a more complete landscape of metabolic pathway alterations across the datasets reveals further consistent, although weaker changes that also involves up-regulation (based on GSEA normalized enrichment score, i.e. NES; Fig. 1E; Table S2; Methods). We note that, however, the apparent directions of these pathway-level changes can be misleading and require careful gene-level or metabolic flux-level inspection (e.g. see the TCA cycle result in the next section). These results suggest that besides immune response, metabolic reprogramming represents one of the most robust changes induced by SARS-CoV-2 infection across various systems, consistent with the key roles of metabolism in viral infection. We next focused on investigating the SARS-CoV-2-induced metabolic changes in the infected host cells.

### Genome-scale metabolic modeling (GEM) identifies SARS-CoV-2-induced patterns of metabolic flux changes

Since gene expression does not necessarily correlate with protein level or enzyme activity and thus may not truthfully reflect metabolic activity (Maier et al. 2009), we applied GEM to infer the metabolic fluxes (i.e. rates of all metabolic reactions) across the datasets. Specifically, for each dataset, the iMAT algorithm (Shlomi et al. 2008) was applied to the median expression profiles of the control and virus-infected samples to compute the refined metabolic models representative of the two respective groups, and the flux distribution of each metabolic reaction was obtained by sampling (Methods). The flux distributions of the control and infected groups were compared and reactions with differential fluxes (DF) were identified (Methods; Table S4). We again examined the consistency across the datasets, here on the flux level, by checking the overlap of the top DF reactions between each pair of datasets. Like on the gene expression level, we are assured by the overall high level of coherence of the DF reactions (odds ratio median 2.40, maximum 28.13; adjusted P value median and minimum both <2.22e-16; Fig. 2A shows the result for the positive DF reactions, similar result for negative DF reactions is in Fig. SX; We note that the sign of DF represents the direction of flux change with regard to the positive direction of a reaction, which can be reversible, and not the increase or decrease of the absolute flux). Although no reaction shows fully consistent changes across all 12 datasets, we identified a set of most consistently changed reactions across datasets while giving higher importance to the in vivo patient datasets (Table S5A; Methods), and examined the metabolic pathways they are enriched in with Fisher’s exact tests (significant pathways with FDR<0.1 shown in Fig. 2B; Table S5B). We see that consistent flux changes are found in various noteworthy pathways including metabolite transport (mitochondrial and extracellular), pentose phosphate pathway, hyaluronan metabolism, pyrimidine biosynthesis, glycine, serine, and threonine metabolism, TCA cycle, inositol phosphate metabolism, fatty acid elongation, among others. Many of these pathways have been implicated in the infection and life cycle of different viruses including SARS-CoV-2 (Mayer et al. 2020, Ou et al. 2020, Gardinassi et al. 2020, Li et al. 2020, Thomas et al. 2020, Ehrlich et al. 2020, Bojkova et al. 2020; see Discussion). Below we closely inspect the flux alteration patterns within these pathways with specialized methods.

First we focused on the cellular transport pathway, seeking to identify the specific transport pattern summarized for each metabolite species (Methods; Table S5C). Some of the metabolites showing the most consistent patterns across datasets are summarized in Fig. 2C. In addition to biological variation, technical noise from the modeling can also contribute to the lack of full consistency in these results. Nevertheless, several glucogenic amino acids, including alanine, asparagine, and glutamate are seen to have increased cellular import from the extracellular space to the cytoplasm (or decreased export) in the virus-infected vs control group across more than half of the datasets. On the other hand, nicotinamide and tetrahydrofolate show relatively consistent patterns of decreased cellular import (or increased export; Fig. 2C). Next, we inspected the other pathways not related to metabolite transport by visualizing their alteration patterns overlaid on the metabolic network, for virus-infected vs the control group. For example, the pyrimidine (de novo) biosynthesis pathway contains consistently increased fluxes towards the synthesis of UMP (the precursor of pyrimidines; Fig. 2D), consistent with the nucleic acid synthesis needs of the virus. In the TCA cycle, we mostly see increased flux driving the normal cycle, together with the interconversion and mitochondrial transport of the cytoplasmic counterparts of TCA cycle intermediates (Fig. 2E). This seems contradictory to the consistent gene expression-level decrease of the “TCA cycle and respiratory electron transport" pathway (Fig. 1D). However, upon closer inspection, we find that the gene-level enrichment of this pathway is mostly driven by the decrease in electron transport chain genes (Table S2), and consistently OXPHOS reactions show decreased fluxes (Table S5A). As examples of pathways with more complex flux change patterns, in the inositol phosphate metabolism pathway, we see increased fluxes converging to phosphatidylinositol 4,5-bisphosphate (pail45p_hs[c]) and inositol (inost[c]), but decreased fluxes to inositol 1-phosphate (mi1p_DASH_D[c]; Fig. 2F); in the glycine, serine, and threonine metabolism pathway, we see decreased conversion of serine to glycine but increased serine degradation to pyruvate (Fig. 2G); in the fatty acid elongation pathway, we see that the synthesis and interconversion of different fatty acids show distinct flux changes (Fig. 2H). These highly intricate metabolic programs revealed by the GEM analysis are consistent with many previous reports and possibly reflect the specific metabolic demands of SARS-CoV-2 during its life cycle (see Discussion), which also demonstrates the value of the modeling approach over gene expression-level analyses.
    
### Prediction of anti-SARS-CoV-2 targets that act via counteracting the virus-induced metabolic changes

We have demonstrated that SARS-CoV-2 can induce recurrent and complex alterations in host cell metabolism. As was proposed previously, targeting the virus-induced metabolic changes can be an effective antiviral strategy (Mayer et al. 2019), which we adopted here to predict anti-SARS-CoV-2 targets. Specifically, we applied the GEM-based rMTA algorithm (Valcárcel et al. 2019) to each of our collected datasets to predict metabolic reactions whose knockout (KO) can transform the cellular metabolism from the SARS-CoV-2-infected state to the non-infected normal state (Methods; Table S6A). MTA computes a score for each of the metabolic reactions in the cell, and usually the 10-20% reactions with the highest MTA score contain promising candidate targets (Yizhak et al. 2013). We first compared the top 10% MTA-predicted reactions across datasets and found that they have reasonable overlap (odds ratio median 3.42, maximum 16.47, Fisher’s exact test adjusted P median and minimum both<2.2e-16 across all pairs of datasets; Fig. 3A). Interestingly, some strong overlaps are seen between certain cell line and patient datasets, consistent with the recurrent metabolic changes across these datasets as seen above.

To validate these predictions, we collected multiple “validation sets” of reported anti-SARS-CoV-2 gene targets or drugs identified from large-scale chemical or genetic screens. These include CRISPR-Cas9 genetic screens in Vero E6 cells (Wei et al. 2020) and cells with exogenous ACE2 expression (A549ACE2, Daniloski et al. 2020), and additional lists of experimentally validated drugs reported in different studies compiled by Kuleshov et al. 2020 (Methods). We first tested for significant overlap between our top 10% MTA-predicted targets from each of the datasets and the validation sets described above with Fisher’s exact tests (after mapping all validated target genes or drugs to the metabolic reactions; Methods). Strongly significant overlaps were found between our predictions from 8 out of the 12 datasets with the antiviral hits (i.e. those whose KO inhibits SARS-CoV-2 infection) identified in the CRISPR-Cas9 screens (FDR<3.95e-4; Fig. 3B), these significant datasets include the Vero (Riva et al. 2020) and A549ACE2 data (Blanco-Melo et al. 2020) from the same cell types as those used in the CRISPR-Cas9 screens, but encouragingly also include three in vivo patient datasets (Lieberman et al. 2020, Liao et al. 2020, Chua et al. 2020). Further examining the experimentally validated anti-SARS-CoV-2 drug sets from previous studies (compiled by Kuleshov et al. 2020), we also found a few cases of significant overlap (FDR<0.1; Fig. 3C), although these drug sets are relatively small, and when we pooled all validated drugs compiled by Kuleshov et al., their targets are only enriched in the predictions from the SC.Chua.Basal dataset (Fig. 3C). Some of our top predicted reactions are also mapped to host proteins identified to interact with SARS-CoV-2 proteins from Gordon et al. 2020 and Stukalov et al. 2020, although we only see weak statistically significant overlaps with the Stukalov et al. data but not the Gordon et al. data (FDR<0.1; Fig. 3D), suggesting that most of our predicted targets are not directly involved in the SARS-CoV-2 life cycle. Overall, our MTA-based top predictions obtained strong validation from the published CRISPR-Cas9 screens, with further but weaker support from the drug screens and host-virus protein-protein interaction (PPI) data.

We further take advantage of the genome-wide CRISPR-Cas9 screens to more closely evaluate the performance of our MTA predictions. Unlike in many of the drug-screen datasets where the screens are low-throughput or complete screen results were not available, we were able to confidently define positive and negative sets (i.e. genes whose KO inhibits or promotes the viral infection, respectively) from the CRISPR-Cas9 screen data. The positive and negative sets were defined in a balanced way (Methods), with which we performed ROC curve analysis of our MTA predictions from each of the dataset (Methods). Although the MTA prediction is only based on the transformation of cellular metabolic states and do not consider the possible effect of other anti-/pro-viral mechanisms, we see that the predictions based on 6 of the datasets achieved area under ROC curve (AUROC) values above 0.6 and as high as 0.81, although two of the other datasets apparently yielded AUROC significantly lesser than 0.5 (Fig. 3E; see Discussion). As examples, ROC curves from the two best-performing datasets are shown in Fig. 3F, we note that the steep proportions of the curves showing close to 1 specificity and decent sensitivity indeed correspond to about top 10-20% of the MTA predictions, consistent with our previous experience with MTA. These results testify that our metabolism-targeting strategy using the MTA algorithm is able to achieve reasonable prediction performances.
Next, we seek to integrate our predictions from the 12 datasets into a final consensus list of high-confidence candidate targets for further extensive experimental validation and investigation. We applied an procedure to pick highly recurrent top predictions across both the in vitro and in vivo datasets (Methods), resulting in a final list of 66 candidate target metabolic reactions of which 59 are mapped to genes and 15 are targeted by known drugs (Table S6B). This final list of candidates are also strongly enriched for the positive targets identified in the two anti-SARS-CoV-2 CRISPR-Cas9 screens described above (Wei et al. 2020 and Daniloski et al. 2020; odds ratio=26.56, P<2.2e-16). These candidates are enriched for metabolic pathways including cellular transport, inositol phosphate and aminosugar metabolism, and within the cellular transport pathway, transporters for various amino acids, succinate, citrate and adenosine are enriched (FDR<0.1; Fig. 3G; Table S6C; Methods). These are consistent with the known biology of SARS-CoV-2, e.g. phosphoinositides are known to be critical for SARS-CoV-2 cell-entry by endocytosis, and inhibiting phosphatidylinositol-3,5-bisphosphate with the drug apilimod has been shown to suppress SARS-CoV-2 entry (Ou et al. 2020).

### Prediction of metabolic targets for anti-SARS-CoV-2 in combination with remdesivir

Given that our MTA-based prediction of single anti-SARS-CoV-2 metabolic targets has yielded promising results, we proceed to extend the same strategy for the prediction of targets that can be combined with remdesivir to achieve higher antiviral efficacy. To this aim, we cultured Vero E6 cells infected by SARS-CoV-2, with or without remdesivir treatment. A control group (no viral infection or remdesivir treatment) a remdesivir-only group (no viral infection) were also included (Methods). Bulk RNA-seq was performed to obtain the gene expression profiles of these samples (Methods). Visualizing the gene expression data with a PCA plot, we see that remdesivir can indeed effectively reverse the virus-associated expression changes (mostly along the first PC axis), but also results in additional orthogonal changes along the second PC axis (Fig. 4A). Performing a GSEA analysis comparing the virus+remdesivir group to the normal control group, we see that many pathways show significant differences in their expression, including some metabolic pathways, e.g. cholesterol and steroid biosynthesis (Fig. 4B; Table S7; Methods). Some of these differences can be attributed to the incomplete reversion of virus-induced expression changes by remdesivir, while others may arise from remdesivir-specific effects (Fig. 4B). Further computing the metabolic flux profiles representative of each group of samples with iMAT (Shlomi et al. 2008) then inspecting the flux-level PCA plot (Fig. 4C; Methods), we observe a similar pattern from that seen on the gene expression level. The differential fluxes between the virus+remdesivir and the control group are enriched for various metabolic pathways (FDR<0.1; Fig. 4D; Table S8; Methods), many also have differential fluxes comparing virus-infected samples to control (see Fig. 2B), suggesting that these metabolic changes are not fully reversed to normal by remdesivir. We hypothesize that further reversing the cellular state in the virus+remdesivir group towards the healthy control state may be an effective combinatory targeting strategy to improve the antiviral efficacy of remdesivir.
    
As before, we focused on the domain of cell metabolism and applied MTA on our data from Vero E6 cells to predict targets for reversing the metabolic flux profile in the virus+remdesivir group towards normal (Methods; Table S9A). Trying to validate these predictions, we obtained a list of 20 experimentally tested drugs showing synergistic anti-SARS-CoV-2 effects with remdesivir in the Calu-3 cell line (Nguyenla et al. 2020). Despite the cell type difference, we observed that the targets of these drugs are significantly enriched by our top 20% MTA predictions (Fisher’s exact test P=0.011, odds ratio 4.83; there is also sign of enrichment by the top 10% MTA predictions with odds ratio 2.01, but it failed to achieve statistical significance at P=0.30). The top 20% predictions recovered 6 of the 11 metabolic reaction targets from Nguyenla et al., corresponding to drugs including cilostazol, ezetimibe, ivosidenib and valdecoxib. Some of the top predicted targets overlap with our predicted single anti-SARS-CoV-2 targets as described above, e.g. various inositol phosphate metabolism reactions. This is consistent with the observation that the virus-induced metabolic changes in these pathways are not effectively reversed to normal by remdesivir (as seen from Fig. 2B and 4D). Performing a pathway enrichment analysis, we see that these top predictions are enriched in bile acid, IMP, and heme biosynthesis pathways, among others (Fig. 4E; Table S9B); cholesterol and steroid metabolism pathways, which were seen to be different on the gene expression level between the virus+remdesivir and control samples (see Fig. 4B), are also enriched for the top predicted targets (Fig. 4E). We provide the top 20% predictions ranked by the MTA score and known drugs for these targets in Table S9A. These predictions represent candidate targets that can potentially improve the antiviral efficacy of remdesivir in combination and warrant further testing in future studies.

## Materials and Methods

### DE analysis

We obtained each of the gene expression datasets on SARS-CoV-2 infection from the sources listed in Table 1. For the bulk RNA-sequencing (RNA-seq) datasets whose read count data is available at the time of analysis, we performed DE analysis comparing the SARS-CoV-2-infected or positive samples to the non-infected control or negative samples with DESeq2 (Love et al. 2014). For Weingarten-Gabby et al. 2020, a nested design for DE was needed and DESeq2 failed to run properly, and limma-voom (Law et al. 2014) was used. For Butler et al. 2020 and Xiong et al. 2020, we obtained the DE results provided from the supplementary materials of the respective publication, as we were not able to obtain their gene-level expression data at the time of analysis. For Butler et al. 2020, among their multiple versions of DE results we used the one from limma-voom with sva correction “Voom:Positive_vs_Negative:10M_samples:sva_correction_2sv”. We also took the DE results of the proteomic data from Bojkova et al. 2020 as provided by the authors, and used the 24 hours post-infection data which is the latest time point available with the largest number of DE proteins). For the single-cell RNA-sequencing (scRNA-seq) datasets, The “FindMarkers” function in the R package Seurat (Stuart et al. 2019) was used to call the MAST method (Finak et al. 2015) for DE analysis in each annotated cell type, with “logfc.threshold” set to 0 to obtain full results across genes. We focused on the airway epithelial cells since our major aim in this study is to investigate the changes in the cell types infected by the SARS-CoV-2 virus, these include the “Epithelial” cell type from Liao et al. 2020 and the “Ciliated” and “Basal” cell types from Chua et al. 2020 (other epithelial subtypes from these datasets yielded no significant DE genes).

### GSEA analysis of the differential expression results

Using the DE log fold-change values from each dataset, GSEA (Subramanian et al. 2005) was performed using the implementation in the R package fgsea (Korotkevich et al. 2019). The gene set/pathway annotations used were the Reactome (Jassal et al. 2020) and KEGG (Kanehisa et al. 2021) subsets from the “Canonical Pathway” category in version 7.0 MSigDB database (Liberzon et al. 2011). For metabolic pathways (in Figure 1E), those under the category “Metabolism” from KEGG (Kanehisa et al. 2021) were used.

### Comparison of the differentially expressed genes and pathways across datasets

The DE results across datasets were compared in a descriptive manner. As a first approach, the DE log fold-change values were inverse normal-transformed across all genes within each dataset, which preserves only the order (i.e. rank) of DE effect sizes, and then PCA was applied to the transformed data. As a second approach, top significantly DE genes or enriched pathways from each pair of datasets were tested for significant overlap using Fisher’s exact tests. For DE genes, the number of significant genes vary greatly across datasets based on a single FDR cutoff, so we uniformly took the top n=400 DE genes ranked by FDR from each datasets (all cases have at least FDR<0.1). For enriched pathways we simply took those with FDR<0.1 from each dataset. To identify the consistent DE changes across datasets, a formal meta-analysis of all 12 datasets is challenging given the wide range of assay platforms and DE algorithms used. So instead, we adopted a subjective criteria that give high importance to the various in vivo patient datasets, such that the results may be more clinically relevant: we identified pathways that are significantly (FDR<0.1) enriched in the consistent direction (up/down-regulation) in at least one of the bulk RNA-seq patient datasets and also at least one of the scRNA-seq datasets, while never showing significant enrichment (FDR<0.1) in the opposite direction in any of the datasets (for the results in Figure 1D; Table S3).

### Computation of metabolic fluxes from gene expression data with GEM

For each dataset, we used the GEM algorithm iMAT (Shlomi et al. 2008) to compute the metabolic flux profile from gene expression data. iMAT requires gene-length normalized expression values in the bulk RNA-seq datasets, for this we computed TPM values with Salmon (Patro et al. 2017) from the raw fastq files for datasets where TPM data is not provided. Then for each dataset, we took the median expression values of the control and virus-infected samples respectively as the representative expression profile for each group, and used it as input to iMAT. The human genome-scale metabolic model (GEM) Recon 1 (Duarte et al. 2007) was used as the base model to iMAT. The main reason we used the older human GEM version was due to the significantly longer computation time and resources needed if later larger models were used instead. The output of iMAT is a refined GEM for the each of the virus-infected and control groups in each dataset, with metabolic reaction bounds adjusted to achieve maximal concordance with the gene expression data while satisfying the stoichiometric constraints of the cellular metabolic network (Shlomi et al. 2008). Each output model defines a space rather than a single unique solution of the global metabolic flux profile, and artificial centering hit-and-run (ACHR) was used to sample the metabolic space and obtain the flux distribution of each metabolic reaction in each condition (control or virus-infected) and dataset. All GEM analyses were performed using our in-house R package named gembox, with the academic version of IBM ILOG CPLEX Optimization Studio 12.10 as the optimization solver.

### Differential flux analysis of virus-infected vs control group in each dataset

The flux distributions of the control and infected groups were compared to identify reactions with differential fluxes (DF). Since an arbitrarily large number of sample points can be sampled from the metabolic space of each group, resulting in statistical tests with arbitrarily small P values, we adopted the following effect size-based criterion for DF reactions: absolute rank biserial correlation (an effect size measure of the difference between the two flux distributions in the control and virus-infected groups) >0.1, and absolute relative flux change (i.e. the absolute difference of the mean fluxes between the two groups over the absolute mean flux in the control group) >5%. Positive DF reactions have flux value difference in infected vs control group >0, and vice versa for negative DF reactions. Note that for non-reversible reactions, flux values are non-negative and the sign of DF can be interpreted similarly to differential gene expression; for reversible reactions, flux values can be negative, representing reactions happening in the reverse direction, thus the sign of DF needs to be interpreted differently, e.g. negative DF represents flux shift towards the reverse direction and not necessarily decrease in absolute flux.

### Analysis of reactions with consistent differential fluxes across datasets and their pathway enrichment analysis

To compare the DF results across datasets, the DF reactions from each pair of datasets were tested for significant overlap using Fisher’s exact tests (separately for positive and negative DF). Since no reaction shows fully consistent DF across all 12 datasets analyzed, similarly as with the DE analysis, we identified the DF reactions with high level of consistency especially in the in vivo patient datasets, such that the results may be more clinically relevant: we identified DF reactions in the consistent direction (positive/negative) in at least one of the bulk RNA-seq patient datasets and also at least one of the scRNA-seq datasets, while showing DF in the opposite direction in no more than 3 datasets, resulting in 293 positive and 349 negative DF reactions (Table S5A). The metabolic pathway enrichment of these DF reactions was analyzed with Fisher’s exact tests (results in Table S5B), with the “subSystems” slot in the metabolic model used as pathway annotation. Note that due to the special interpretation of the sign of DF values as explained above, the GSEA used for gene expression-level analysis is not appropriate for pathway enrichment analysis on the flux level.

### Differential transport flux analysis by each metabolite species

Differential transport flux analysis on the metabolite level was first performed for each dataset. Specifically, in each group (control or virus-infected), the sampled flux values on the reaction level for all transport reactions across the cell membrane were summarized by each metabolite, taking the direction of transport (i.e. import or export) into consideration to produce the net cellular transport flux of the metabolite (positive means net import, vice versa). The metabolites with differential net transport fluxes between the control and virus-infected groups were identified as in the reaction-level DF analysis described above. The consistent DF metabolites were then identified in the same way as with the DF reactions described above.

### Analysis of the consistent flux alteration patterns in different metabolic pathways

For each of the significantly enriched metabolic pathways identified in the consistent DF reaction analysis described above, we defined the “consensus” direction of each reaction as represented by those shown in the virus-infected group from the majority (>6 out of 12) of the datasets, and also similarly for the “consensus” direction of DF for each reaction. The consensus directions of reactions and their DF were overlaid onto network diagrams of the pathways and visualized, where metabolites are represented by nodes, reactions are represented by directed (hyper) edges with edge direction corresponding to the consensus reaction direction and edge color corresponding to the consensus DF direction. Parts of the metabolic pathways where reactions are not consistently altered across datasets are greyed out or removed to increase the clarity. Upon visual inspection, potential futile loops in the network are also removed from the visualizations.

### Prediction of anti-SARS-CoV-2 target metabolic reactions with metabolic transformation algorithm

For each of the collected datasets, the DE result of virus-infected vs control samples as well as the representative flux distribution of the virus-infected group computed with iMAT (Shlomi et al. 2008) followed by ACHR sampling were used as inputs for the GEM-based metabolic transformation algorithm (MTA; Yizhak et al. 2013; a variant called rMTA was used; Valcárcel et al. 2019) to predict metabolic reactions whose knock-out can transform cellular metabolic state from that of the virus-infected to that of the control samples (full prediction results from all datasets in Table S6A). The output of rMTA is a score (rMTA score) for each metabolic reaction, with higher scores corresponding to better candidates for achieving the metabolic transformation as specified above. From our previous experience (Yizhak et al. 2013), the top 10-20% MTA predictions contain promising targets. The human Recon 1 GEM (Duarte et al. 2007) was used for the MTA analysis. The rMTA algorithm implemented in our in-house R package named gembox was used, with the academic version of IBM ILOG CPLEX Optimization Studio 12.10 as the optimization solver. To compare the MTA predictions across datasets, the top 10% predictions from each pair of datasets were tested for significant overlap using Fisher’s exact tests.

### Validation of the MTA-predicted anti-SARS-CoV-2 metabolic targets

Multiple datasets of reported anti-SARS-CoV-2 gene targets or drugs identified from large-scale chemical or genetic screens were collected to validate our predictions. Gene-level results of two published CRISPR-Cas9 genetic screens (Wei et al. 2020 and Daniloski et al. 2020) were obtained from the supplementary materials of the respective publication. For Wei et al., gene hits with FDR<0.1 and mean z score>0 (i.e. KO inhibits the viral infection) were taken; Daniloski et al. reported two screens with different multiplicities of infections (MOIs) and provided only single-sided FDR, so gene hits with FDR<0.1 from either screen were taken. The union set of hits from both studies were used (Table S6D). Lists of experimentally validated drugs reported in different studies compiled by Kuleshov et al. 2020 were downloaded from https://maayanlab.cloud/covid19/, which are then mapped to the genes they inhibit using data from DrugBank v5.1.7 (Wishart et al. 2018). Additionally, host proteins identified to interact with SARS-CoV-2 proteins were obtained from the supplementary materials of Gordon et al. 2020 and Stukalov et al. 2020. The genes from these validation datasets are mapped to metabolic reactions wherever applicable based on the human GEM Recon 1 (Duarte et al. 2007) data. Then, the significant overlap between the top 10% MTA-predicted targets from each dataset and each of the validation sets described above was tested with Fisher’s exact tests on the reaction level (full results in Table S6E). Reaction-level test is performed because multiple reactions can be mapped to the same gene, and performing Fisher’s exact test on the gene-level fails to consider such multiple mapping and is thus inappropriate.

For ROC analysis, negative sets (i.e. genes whose KO promotes SARS-CoV-2 infection) were defined based on the two CRISPR-Cas9 screens described above. For Wei et al., gene with FDR<0.1 and mean z score<0 were taken; since Daniloski et al. provided only single-sided FDR, the log fold-change threshold corresponding to the FDR<0.1 cutoff was identified, and genes with more extreme log fold-changes in the opposite direction were taken. The union of the negative sets from both studies was used. Both the positive (described in the previous paragraph) and negative sets of genes are then mapped to metabolic reactions as described above. The negative set defined as such contains a relatively balanced number of reactions compared to the positive set (131 vs 81). The rMTA score for the reactions produced by MTA was used as the predicted value for ROC analysis. The R package pROC (Robin et al. 2011) was used to compute the AUROC values and their 95% confidence intervals (the latter computed with bootstrapping).

### Defining and analyzing the consensus set of candidate anti-SARS-CoV-2 metabolic targets across datasets

Based on top 10% MTA predictions from the 12 datasets (6 in vitro and 6 in vivo), the metabolic reaction targets that are recurrent in at least 3 of the in vitro datasets, and also in 3 of the in vivo datasets (i.e. the intersection of the two) were taken to be the final consensus candidate targets with high-confidence support across datasets. These target reactions are also mapped to genes using the human GEM Recon 1 data (Duarte et al. 2007), and further mapped to known drugs inhibiting the gene targets using data from DrugBank v5.1.7 (Wishart et al. 2018; target list given in Table S6B). The metabolic pathway enrichment of these targets was analyzed with Fisher’s exact tests, with the “subSystems” slot in the metabolic model used as pathway annotation. For enrichment analysis of cellular transport reactions by each metabolite species, the reactions responsible for transporting each metabolite across the cell membrane were identified from Recon 1 and used as reaction sets for the Fisher’s exact tests.

### Preparation of Vero E6 cell samples with SARS-CoV-2 infection and remdesivir treatment, RNA-sequencing, and gene expression data analysis

Vero E6 cells (ATCC® CRL-1586™) were maintained in Dulbecco’s modified eagle medium (DMEM, Gibco) supplemented with 10% heat-inactivated fetal bovine serum (FBS, Gibco), 50 U ml−1 penicillin, 50 μg ml−1 streptomycin, 1 mM sodium pyruvate (Gibco), 10 mM 4-(2-hydroxyethyl)-1-piperazineethanesulfonic acid (HEPES, Gibco), and 1× MEM non-essential amino acids solution (Gibco). The SARS-CoV-2 USA-WA1/2020 strain was obtained from BEI Resources (NR-52281). The virus was inoculated on Vero E6 cells and the cell supernatant was collected at 72 h post-inoculation (hpi), when extensive cytopathic effects were observed. The supernatant, after clarification by centrifugation 15 min at 4 °C at 5,000 xg, was aliquoted and stored at −80 °C until use. 500,000 Vero E6 cells were seeded in 6-well plates. The following day, the cell medium was replaced with fresh medium supplemented with either DMSO or 1 µM remdesivir (Adooq Bioscience), and cells were either mock-infected or infected with SARS-CoV-2 USA-WA1/2020 (MOI=0.3). Twenty-four hours after infection, cells were collected, and total intracellular RNA was extracted using the Qiagen® RNeasy® Plus Mini Kit. Three replicates were performed for each group, resulting in a total of six samples. The quality of the extracted RNA was assessed with Agilent® 2100 Bioanalyzer. Libraries were prepared on total RNA following ribosome RNA depletion with standard protocol according to Illumina®. Total RNA sequencing was then performed on the Illumina® NextSeq system, 150bp paired-end runs were performed and 100 million raw reads per sample were generated. STAR (Dobin et al. 2013) was used to align the reads to reference genome of the African green monkey (Chlorocebus sabaeus, https://useast.ensembl.org/Chlorocebus_sabaeus/Info/Annotation), with the SARS-CoV-2 genome (https://www.ncbi.nlm.nih.gov/nuccore/NC_045512) added to the reference genome. DESeq2 (Love et al. 2014) was used for DE analysis between pairs of experimental groups (including virus+remdesivir vs control and virus vs control; DE results in Table S7). GSEA (Subramanian et al. 2005) was performed using the implementation in the R package fgsea (Korotkevich et al. 2019), results are provided in Table S7. The gene set/pathway annotations used were the Reactome (Jassal et al. 2020) and KEGG (Kanehisa et al. 2021) subsets from the “Canonical Pathway” category in version 7.0 MSigDB database (Liberzon et al. 2011).

### Genome-scale metabolic modeling of the remdesivir-treated Vero E6 cell samples and prediction of anti-SARS-CoV-2 metabolic targets in combination with remdesivir

As with the metabolic modeling of the other datasets on SARS-CoV-2 infection, iMAT (Shlomi et al. 2008) together with ACHR was used to compute the metabolic flux distribution for each of the experimental groups, using the median expression TPM values of each group as the input to iMAT. Reactions with differential fluxes (DF) between groups (including virus+remdesivir vs control and virus vs control) were identified as described above, and their significant metabolic pathway enrichment was tested with Fisher’s exact tests, with pathways defined by the “subSystems” from the Recon 1 model (Duarte et al. 2007; results in Table S8). The DE result of virus+remdesivir vs control group and the mean flux distribution of the virus+remdesivir group computed with iMAT were used as inputs for MTA to predict metabolic reactions whose knock-out can further transform the virus+remdesivir metabolic state to the normal control state (results in Table S9A). The top 10% and 20% MTA-predicted targets were tested for significant enrichment for the targets of a list of experimentally validated synergistic drugs with remdesivir (Nguyenla et al. 2020) using Fisher’s exact test (performed on the metabolic reaction level as described above). Metabolic pathway enrichment analysis of the top MTA-predicted targets was performed as described above (results in Table S9B).

### Softwares and code

R version 3.6.3 was used for all statistical tests. P values lesser than 2.22e-16 may not be computed accurately and are reported as “P<2.22e-16” throughout the text. The Benjamini-Hochberg (BH) method was used for P value adjustment throughout the text. The R packages ggplot2 (Wickham 2016), ComplexHeatmap (Gu et al. 2016) and visNetwork (https://cran.r-project.org/web/packages/visNetwork/index.html) were used to create the visualizations. The code used for the analyses can be found in the GitHub repository: https://github.com/ruppinlab/covid_metabolism.

## Discussion and Conclusion

text
  
