```{r, include=FALSE}
library(my.utils)
library(gembox)
library(ComplexHeatmap)
```


# Modeling of host metabolism during SARS-CoV-2 infection ^[The work of this chapter has been written into a manuscript and submitted to the online preprint server bioRxiv (@cheng_genome-scale_2021); it has also been submitted to a journal, which is under peer-review as of Mar 2020. Many contents of this chapter are taken from the above article, involving direct quotes with minor rephrasing. This study involves biological experimental work performed by Dr. Laura Riva in Dr. Sumit K. Chanda's lab at SBP, with whom we collaborated.] {#covid}

## Introduction

The COVID-19 pandemic has been ongoing for a year as of March 2021. Despite unprecedented global efforts to control the pandemic, this health crisis has not been resolved to date. While the wide deployment of vaccination is expected to significantly slow down the spread of the disease, the evolution and emergence of variant virus strains suggests a possible need to update the vaccines in the future [@wang_mrna_2021]. Some studies predicted that there may be recurrent wintertime COVID-19 outbreaks over a longer term after the initial pandemic [@kissler_projecting_2020]. Currently, effective therapeutic options are still very scarce, and thus the development of new treatment strategies are still urgently needed for long-term control of the pandemic (Section \@ref(intro-covid-therapy)). Drug repurposing and combination may represent practical and time-efficient strategies to address this urgent unmet medical need. Given the recognized importance of host metabolism in coronavirus (including SARS-CoV-2) infection and proliferation (Section \@ref(intro-covid-metab)), here I aim to identify anti-SARS-CoV-2 targets that act via counteracting the virus-induced metabolic changes, either individually or in combination with current anti-SARS-CoV-2 drugs, especially remdesivir. Such a metabolic-targeting approach may be especially valuable for anti-SARS-CoV-2 drug repurposing to address the current urgent COVID-19 crisis considering that many existing drugs are metabolism-targeting. My specific approach involves the application of genome-scale metabolic modeling (GEM) in an integrated analysis of multiple published *in vitro* and human patient gene expression datasets on SARS-CoV-2 infection, in order to comprehensively identify SARS-CoV-2 induced host metabolic reprogramming. Then I applied the improved version of the metabolic transformation algorithm (MTA) to predict the metabolic anti-SARS-CoV-2 targets. The predictions were validated with the experimentally identified anti-SARS-CoV-2 drugs and targets reported in published drug or genetic screens.

## Results

### Analysis of SARS-CoV-2-induced gene expression changes across datasets

Taking advantage of multiple published studies on SARS-CoV-2 infection, I collected a total of 12 published relevant datasets involving different sample types including various cell lines, primary bronchial epithelial cells, nasopharyngeal swab and bronchoalveolar lavage fluid (BALF) from patients, and also based on different assay platforms including bulk RNA-seq, scRNA-seq, and MS-based proteomics. These datasets are summarized in Table \@ref(tab:covid-datasets). I performed differential expression (DE) analysis of the SARS-CoV-2-infected samples vs the non-infected controls. For the single-cell datasets, only the airway epithelial cells are used, as they are known as the major virus-infected cell type. The PCA plot of the datasets based on the inverse normal-transformed DE log fold-change values is shown in Figure \@ref(fig:covid-de-pca).

```{r}
covid.datasets <- fread("data/covid/table1.tsv")
```

```{r covid-datasets, results="asis"}
knitr::kable(covid.datasets, format="latex",
  caption = "Summary of the published gene expression datasets on SARS-CoV-2 infection analyzed in this study",
  #longtable = TRUE,
  booktabs = TRUE
) %>% kableExtra::kable_styling(latex_options="scale_down")
```

```{r covid-de-pca, out.width="50%", fig.cap="PCA plot of the datasets based on the inverse normal-transformed DE log fold-change values."}
knitr::include_graphics(path="figure/covid/de.pca.svg.png")
```

The PCA plot suggests that the cell lines tend to have distinct DE profiles from the patient samples, although different patient datasets of different sample types and sequencing platforms exhibit considerable variation. Similar pattern is found by comparing each pair of datasets for their top significant DE genes (using Fisher's exact tests, top n=400 DE genes from each datasets were used, all with at least FDR<0.1), as shown in Figure \@ref(fig:covid-de-mat). Across all pairs of comparisons, the median odds ratio value is 2.18 and maximum value is 13.60 (adjusted P median 5.09e-5, minimum<2.22e-16), representing reasonable similarities across datasets.

```{r covid-de-mat, out.width="80%", fig.cap="Pairwise comparison of the datasets by their top significant DE genes with Fisher's exact tests."}
knitr::include_graphics(path="figure/covid/de.enr.mat.svg.png")
```

I then performed gene set enrichment analysis (GSEA) [@subramanian_gene_2005] based on the DE results of each dataset. The level of coherence across datasets on the pathway level appears stronger (Appendix Figure \@ref(fig:covid-gsea-mat)), with the median odds ratio from all pairwise comparison using Fisher's exact test equals to 4.25 (maximum is infinity followed by 35.49), with adjusted P median 2.01e-5 (minimum<2.22e-16). No pathway is consistently significant across all 12 datasets. I identified the most consistently enriched pathways across the datasets, especially with emphasis the various in vivo patient datasets, and the resulting pathways are visualized in Figure \@ref(fig:covid-de-gsea-consensus), Many up-regulated pathways are involved in innate immune response to viral infection, e.g. cytosolic DNA sensing and interferon signaling. Antigen presentation via MHC-I is seen coherently down-regulated in SARS-CoV-2 infection, and other down-regulated pathways span many major categories of cellular metabolism, e.g. OXPHOS (energy metabolism), valine, leucine and isoleucine degradation (amino acid metabolism), sphingolipid metabolism (lipid metabolism), and N-glycan biosynthesis, etc. These possibly reflect the specific metabolic demands of SARS-CoV-2 or are related to its pathogenic effects. These results also suggest that besides immune response, metabolic reprogramming represents one of the most robust changes induced by SARS-CoV-2 infection across various systems, consistent with the key roles of metabolism in viral infection. However, it should be note that there is no global downregulation of metabolic activity upon SARS-CoV-2 infection, as can be seen in Appendix Figure \@ref(fig:covid-gsea-metab). As metabolic fluxes cannot be directly reflected by the gene expression levels. I therefore next seek to investigate the SARS-CoV-2-induced metabolic changes using GEM methods.

```{r covid-de-gsea-consensus, out.width="100%", fig.cap="Summary of the top consistently enriched pathways from GSEA analysis across the analyzed datasets."}
knitr::include_graphics(path="figure/covid/gsea.enr.recur.mat.svg.png")
```

### GEM-based analysis of SARS-CoV-2-induced metabolic flux change patterns

I next applied the iMAT algorithm [@shlomi_network-based_2008] to the median expression profiles of the control and virus-infected samples to compute the refined metabolic models representative of the two respective groups, and the flux distribution of each metabolic reaction was obtained by random sampling (ACHR method). The reactions with differential fluxes (DF) between the virus-infected samples vs controls were identified. I checked the overlap of the top DF reactions between each pair of datasets with Fisher's exact tests to measure the level of consistency on the flux level. Like on the gene expression level, there is an overall high level of coherence of the DF reactions (odds ratio median 2.40, maximum 28.13; adjusted P value median and minimum both <2.22e-16; see Appendix Figure \@ref(fig:covid-df-pos-mat) and Figure \@ref(fig:covid-df-neg-mat) for the positive and negative DF reactions, respectively). Again, I identified the most consistently altered reactions across datasets with stronger emphasis on the various *in vivo* patient datasets and used Fisher's exact tests to identify the metabolic pathways they are enriched in. The significantly enriched pathways with FDR<0.1 are shown in Figure \@ref(fig:covid-df-path-consensus). Consistent flux changes are seen in many different pathways, including metabolite transport (mitochondrial and extracellular), pentose phosphate pathway, hyaluronan metabolism, pyrimidine biosynthesis, glycine, serine, and threonine metabolism, TCA cycle, inositol phosphate metabolism, fatty acid elongation.

```{r covid-df-path-consensus, out.width="100%", fig.cap="Summary of the metabolic pathways enriched by the metabolic reactions showing the most consistent changes across the analyzed datasets."}
knitr::include_graphics(path="figure/covid/df.enr.svg.png")
```

I then inspected the flux alteration patterns within these pathways more closely. First, I aim to identify the specific patterns in metabolite transportation, summarized by each metabolite. Metabolites with the most consistent patterns across datasets are shown in Figure \@ref(fig:covid-df-tx-consensus). In more than half of the datasets, several glucogenic amino acids, e.g. alanine, asparagine, and glutamate have increased cellular import from the extracellular space to the cytoplasm (or decreased export) in the virus-infected vs control samples, while nicotinamide and tetrahydrofolate show relatively consistent decrease in cellular import (or increase in export).

```{r covid-df-tx-consensus, out.width="100%", fig.cap="Summary of the metabolites showing the most consistent transportation patterns between the extracellular space and cytosol across the analyzed datasets."}
knitr::include_graphics(path="figure/covid/tx.met.mat.svg.png")
```

Next, I visualized the metabolic networks for different individual pathways that are enriched for consistently changed reactions, in order to inspect the patterns of flux changes in these pathways. I show two example visualizations below, the others are given in Appendix. The pyrimidine *de novo* biosynthesis pathway has consistently increased fluxes towards the synthesis of UMP, the precursor of other pyrimidines, consistent with the nucleic acid synthesis requirement of SARS-CoV-2 (Figure \@ref(fig:covid-flux-pyrimidine), red and blue denotes increased and decreased absolute flux values, respectively; grey denotes inconsistent changes). TCA cycle mostly show increased flux in the normal direction of the cycle as well as in the interconversion and mitochondrial transport of the cytoplasmic counterparts of TCA cycle intermediates (Appendix Figures \@ref(fig:covid-flux-tca1) and \@ref(fig:covid-flux-tca2)). This seems contradictory to the consistent gene expression-level decrease of the "TCA cycle and respiratory electron transport" pathway (Figure \@ref(fig:covid-de-gsea-consensus)). However, a closer inspection reveals that the gene-level enrichment is mostly driven by the decrease in ETC genes rather than TCA genes, and consistently OXPHOS reactions show decreased fluxes.

```{r covid-flux-pyrimidine, out.width="80%", fig.cap="Visualization of the consistent flux changes in the pyrimidine biosynthesis pathway. "}
knitr::include_graphics(path="figure/covid/pyrimidine.png")
```

Other pathways show more complex flux change patterns, e.g. the inositol phosphate metabolism pathway contains increased fluxes converging to phosphatidylinositol 4,5-bisphosphate (pail45p_hs[c]) and inositol (inost[c]), but decreased fluxes to inositol 1-phosphate (mi1p_DASH_D[c]; Appendix Figure \@ref(fig:covid-flux-inositol)). In the glycine, serine, and threonine metabolism pathway, there is decreased conversion of serine to glycine but increased serine degradation to pyruvate (Appendix Figure \@ref(fig:covid-flux-glycine)); in the fatty acid elongation pathway, the synthesis and interconversion of different fatty acids show varied flux changes (Figure \@ref(fig:covid-flux-fas)). These results show that SARS-CoV-2 can induce highly complex metabolic alterations potentially to meet their metabolic demands during their intracellular life cycle. They also demonstrate that GEM is able to add to gene expression-level analysis in recovering these fine-grained alteration patterns.

```{r covid-flux-fas, out.width="80%", fig.cap="Visualization of the consistent flux changes in the fatty acid elongation pathway. Red and blue denotes increased and decreased absolute flux values, respectively; grey denotes inconsistent changes."}
knitr::include_graphics(path="figure/covid/fas.png")
```

### Prediction and validation of metabolic anti-SARS-CoV-2 targets with MTA

Based on the proposal that targeting the virus-induced metabolic changes can be an effective antiviral strategy [@mayer_hijacking_2019] (Chapter \@ref(intro-covid-metab)), I next used MTA [described in Chapter \@ref(mta), based on @yizhak_model-based_2013; @valcarcel_rmta_2019] to predict anti-SARS-CoV-2 targets, specifically those metabolic reactions whose KO can revert the cellular metabolic state from that of the virus-infected to that of the normal non-infected samples. As seen in Chapter \@ref(mta), the 10-20% reactions with the highest MTA score usually contain promising candidate targets. I inspected the overlaps of the top 10% MTA-predicted reactions based on each pair of datasets, and found overall reasonable overlaps (odds ratio median 3.42, maximum 16.47, Fisher's exact test adjusted P median and minimum both<2.2e-16 across all pairwise comparison). Interestingly, there are some strong overlaps between certain cell line and patient datasets (Appendix).

I next aimed to validate these predictions using published anti-SARS-CoV-2 gene targets or drugs identified from large-scale chemical or genetic screens. First, I collected the top genes whose KO was found to inhibit SARS-CoV-2 infection in CRISPR-Cas9 genetic screens in Vero E6 cells [@wei_genome-wide_2021] and A549 cells with exogenous ACE2 expression [@daniloski_identification_2021]. The top 10% MTA-predictions from 8 out of the 12 datasets were found to have significant overlaps with these anti-viral hits from the CRISPR screens (after mapping to metabolic reactions, FDR<3.95e-4; Figure \@ref(fig:covid-mta-genetic-enr)). Second, checking additional lists of experimentally validated drugs reported in different studies from [@kuleshov_covid-19_2020], there were also a few cases of significant overlap (FDR<0.1; Figure \@ref(fig:covid-mta-drug-enr)). Third, the top predictions from some of the datasets were also enriched for host proteins identified to interact with SARS-CoV-2 proteins from [@stukalov_multi-level_2020], although not those from [@gordon_sars-cov-2_2020] (Appendix Figure \@ref(fig:covid-mta-ppi-enr)).

```{r covid-mta-genetic-enr, out.width="100%", fig.cap="Summary of the enrichment of top MTA predictions from each dataset in the antiviral gene targets identified from CRISPR-Cas9 screens."}
knitr::include_graphics(path="figure/covid/mta.screen.enr.svg.png")
```

```{r covid-mta-drug-enr, out.width="100%", fig.cap="Summary of the enrichment of top MTA predictions from each dataset in the antiviral drugs identified from various experimental drug screens."}
knitr::include_graphics(path="figure/covid/mta.drug.enr.svg.png")
```

Further, based on the two genome-wide CRISPR-Cas9 screens described above, I defined comprehensive sets of positive and negative genes (i.e. genes whose KO inhibits or promotes the viral infection, respectively). These two sets are defined in a way such that they are balanced. After mapping these to the metabolic reaction level as the validation sets, I performed ROC curve analysis of the MTA predictions. The predictions from 6 of the datasets achieved area under ROC curve (AUROC) values above 0.6 and as high as 0.81, although two of the other datasets apparently yielded AUROC significantly smaller than 0.5 (Figure \@ref(fig:covid-mta-auroc)). As examples, the ROC curve from the best-performing Calu-3 and SC.Liao datasets are shown in Appendix Figures \@ref(fig:covid-mta-roc-calu3) and \@ref(fig:covid-mta-roc-liao). These results demonstrate that our metabolism-targeting strategy using the MTA algorithm is able to achieve reasonable prediction performances.

```{r covid-mta-auroc, out.width="100%", fig.cap="Summary of the AUROC of top MTA predictions from each dataset, based on validated positive and negative hits from CRISPR-Cas9 screens."}
knitr::include_graphics(path="figure/covid/auc.svg.png")
```

I next integrated the predictions from the 12 datasets into a final consensus list of high-confidence candidate targets (Methods), resulting in a final list of 66 candidate target metabolic reactions of which 59 are mapped to genes and 15 are targeted by known drugs. This final list of candidates are also strongly enriched for the positive targets identified in the two anti-SARS-CoV-2 CRISPR-Cas9 screens described above (odds ratio=26.56, P<2.2e-16). These candidates are enriched for cellular transport, inositol phosphate and aminosugar metabolism pathways, and within the cellular transport pathway, transporters for various amino acids, succinate, citrate and adenosine are enriched (FDR<0.1; Figure \@ref(fig:covid-mta-pathway-consensus)).

```{r covid-mta-pathway-consensus, out.width="100%", fig.cap="Summary of the metabolic pathway enrichment result of the consensus MTA-predicted targets across datasets."}
knitr::include_graphics(path="figure/covid/mta.path.enr.svg.png")
```

### Prediction of metabolic targets for anti-SARS-CoV-2 in combination with remdesivir

Given that the MTA prediction of single anti-SARS-CoV-2 targets has yielded promising results, I applied the same strategy for the prediction of targets that can be combined with remdesivir to achieve higher antiviral efficacy. To this aim, we teamed up with experimental collaborators who cultured Vero E6 cells infected by SARS-CoV-2, with or without remdesivir treatment. A control group (no viral infection or remdesivir treatment) a remdesivir-only group (no viral infection) were also included. We then performed RNA-seq on these samples. The PCA plot visualizing the gene expression of these samples are shown in Figure \@ref(fig:covid-rem-expr-pca). From this plot it can be seen that remdesivir can indeed effectively reverse the virus-associated expression changes, but also results in additional orthogonal changes. Pathway enrichment via GSEA comparing the virus+remdesivir group to the control group confirms that there are still significant differences in many pathways, including some metabolic ones, e.g. cholesterol and steroid biosynthesis (Appendix Figure \@ref(fig:covid-rem-expr-pathway)).

```{r covid-rem-expr-pca, out.width="80%", fig.cap="PCA plot on the transcriptome data from control Vero E6 cells as well as SARS-CoV-2-infected cells with or without remdesivir treatment"}
knitr::include_graphics(path="figure/covid/rem.pca.expr.svg.png")
```

Using the iMAT algorithm to compute the metabolic fluxes of the samples as before, I confirmed that remdesivir does not fully reverse cell metabolism back to normal on the flux level (Appendix Figure \@ref(fig:covid-rem-flux-pca)). The metabolic pathways enriched by the differential fluxes between the virus+remdesivir and the control group are visualized in Figure \@ref(fig:covid-rem-flux-pathway). Based on the hypothesize that further reversing the cellular state in the virus+remdesivir group towards the healthy state may be an effective combinatory targeting strategy to improve the antiviral efficacy of remdesivir, I applied MTA to predict targets for reversing these metabolic fluxes. There is little data available on known targets or drugs that are synergistic with remdesivir. As a preliminarly validation of the predictions, I checked the enrichment between the top MTA prediction with 20 experimentally tested drugs showing synergistic anti-SARS-CoV-2 effects with remdesivir in the Calu-3 cell line reported in [@nguyenla_discovery_2020]. There is a significant enrichment using the top 20% MTA predictions (Fisher's exact test P=0.011, odds ratio 4.83; there is also sign of enrichment using the top 10% predictions with odds ratio 2.01, but it failed to achieve statistical significance at P=0.30), recovering 6 of the 11 metabolic reaction targets from [@nguyenla_discovery_2020]. The metabolic pathways enriched by the top MTA predictions include bile acid, IMP, cholesterol and steroid metabolism pathways, and heme biosynthesis pathways, among others (Figure \@ref(fig:covid-rem-mta-pathway)).

```{r covid-rem-flux-pathway, out.width="100%", fig.cap="Summary of the metabolic pathway enrichment of the top differential fluxes between the virus+remdesivir and the control Vero E6 cells"}
knitr::include_graphics(path="figure/covid/enr.c2vd.flux.svg.png")
```

```{r covid-rem-mta-pathway, out.width="100%", fig.cap="Summary of the metabolic pathway enrichment result of the top MTA-predicted targets for combination with remdesivir"}
knitr::include_graphics(path="figure/covid/rem.mta.path.enr.svg.png")
```

## Materials and Methods

### DE analysis

We obtained each of the gene expression datasets on SARS-CoV-2 infection from the sources listed in Table \@ref(tab:covid-datasets) [@riva_discovery_2020; @blanco-melo_imbalanced_2020; @weingarten-gabbay_sars-cov-2_2020; @bojkova_proteomics_2020; @butler_shotgun_2020; @lieberman_vivo_2020; @xiong_transcriptomic_2020; @liao_single-cell_2020; @chua_covid-19_2020]. For the bulk RNA-sequencing (RNA-seq) datasets whose read count data is available at the time of analysis, we performed DE analysis comparing the SARS-CoV-2-infected or positive samples to the non-infected control or negative samples with DESeq2 [@love_moderated_2014]. For [@weingarten-gabbay_sars-cov-2_2020], a nested design for DE was needed and DESeq2 failed to run properly, and limma-voom [@law_voom_2014] was used. For [@butler_shotgun_2020; @xiong_transcriptomic_2020], we obtained the DE results provided from the supplementary materials of the respective publication, as we were not able to obtain their gene-level expression data at the time of analysis. For [@butler_shotgun_2020], among their multiple versions of DE results we used the one from limma-voom with sva correction "Voom:Positive_vs_Negative:10M_samples:sva_correction_2sv". We also took the DE results of the proteomic data from [@bojkova_proteomics_2020] as provided by the authors, and used the 24 hours post-infection data which is the latest time point available with the largest number of DE proteins). For the single-cell RNA-sequencing (scRNA-seq) datasets, The "FindMarkers" function in the R package Seurat [@stuart_comprehensive_2019] was used to call the MAST method [@finak_mast_2015] for DE analysis in each annotated cell type, with "logfc.threshold" set to 0 to obtain full results across genes. We focused on the airway epithelial cells since our major aim in this study is to investigate the changes in the cell types infected by the SARS-CoV-2 virus, these include the "Epithelial" cell type from [@liao_single-cell_2020] and the "Ciliated" and "Basal" cell types from [@chua_covid-19_2020] (other epithelial subtypes from these datasets yielded no significant DE genes).

### GSEA analysis of the differential expression results

Using the DE log fold-change values from each dataset, GSEA [@subramanian_gene_2005] was performed using the implementation in the R package fgsea [@korotkevich_fast_2021]. The gene set/pathway annotations used were the Reactome [@jassal_reactome_2020] and KEGG [@kanehisa_kegg_2021] subsets from the "Canonical Pathway" category in version 7.0 MSigDB database [@liberzon_molecular_2011]. For metabolic pathways, those under the category "Metabolism" from KEGG [@kanehisa_kegg_2021] were used.

### Comparison of the differentially expressed genes and pathways across datasets

The DE results across datasets were compared in a descriptive manner. As a first approach, the DE log fold-change values were inverse normal-transformed across all genes within each dataset, which preserves only the order (i.e. rank) of DE effect sizes, and then PCA was applied to the transformed data. As a second approach, top significantly DE genes or enriched pathways from each pair of datasets were tested for significant overlap using Fisher's exact tests. For DE genes, the number of significant genes vary greatly across datasets based on a single FDR cutoff, so we uniformly took the top n=400 DE genes ranked by FDR from each datasets (all cases have at least FDR<0.1). For enriched pathways we simply took those with FDR<0.1 from each dataset. To identify the consistent DE changes across datasets, a formal meta-analysis of all 12 datasets is challenging given the wide range of assay platforms and DE algorithms used. So instead, we adopted a subjective criteria that give high importance to the various *in vivo* patient datasets, such that the results may be more clinically relevant: we identified pathways that are significantly (FDR<0.1) enriched in the consistent direction (up/down-regulation) in at least one of the bulk RNA-seq patient datasets and also at least one of the scRNA-seq datasets, while never showing significant enrichment (FDR<0.1) in the opposite direction in any of the datasets.

### Computation of metabolic fluxes from gene expression data with GEM

For each dataset, we used the GEM algorithm iMAT [@shlomi_network-based_2008] to compute the metabolic flux profile from gene expression data. iMAT requires gene-length normalized expression values in the bulk RNA-seq datasets, for this we computed TPM values with Salmon [@patro_salmon_2017] from the raw fastq files for datasets where TPM data is not provided. Then for each dataset, we took the median expression values of the control and virus-infected samples respectively as the representative expression profile for each group, and used it as input to iMAT. The human genome-scale metabolic model (GEM) Recon 1 [@duarte_global_2007] was used as the base model to iMAT. The main reason we used the older human GEM version was due to the significantly longer computation time and resources needed if later larger models were used instead. The output of iMAT is a refined GEM for the each of the virus-infected and control groups in each dataset, with metabolic reaction bounds adjusted to achieve maximal concordance with the gene expression data while satisfying the stoichiometric constraints of the cellular metabolic network [@shlomi_network-based_2008]. Each output model defines a space rather than a single unique solution of the global metabolic flux profile, and artificial centering hit-and-run (ACHR) was used to sample the metabolic space and obtain the flux distribution of each metabolic reaction in each condition (control or virus-infected) and dataset. All GEM analyses were performed using our in-house R package named gembox, with the academic version of IBM ILOG CPLEX Optimization Studio 12.10 as the optimization solver.

### Differential flux analysis of virus-infected vs control group in each dataset

The flux distributions of the control and infected groups were compared to identify reactions with differential fluxes (DF). Since an arbitrarily large number of sample points can be sampled from the metabolic space of each group, resulting in statistical tests with arbitrarily small P values, we adopted the following effect size-based criterion for DF reactions: absolute rank biserial correlation (an effect size measure of the difference between the two flux distributions in the control and virus-infected groups) >0.1, and absolute relative flux change (i.e. the absolute difference of the mean fluxes between the two groups over the absolute mean flux in the control group) >5%. Positive DF reactions have flux value difference in infected vs control group >0, and vice versa for negative DF reactions. Note that for non-reversible reactions, flux values are non-negative and the sign of DF can be interpreted similarly to differential gene expression; for reversible reactions, flux values can be negative, representing reactions happening in the reverse direction, thus the sign of DF needs to be interpreted differently, e.g. negative DF represents flux shift towards the reverse direction and not necessarily decrease in absolute flux.

### Analysis of reactions with consistent differential fluxes across datasets and their pathway enrichment analysis

To compare the DF results across datasets, the DF reactions from each pair of datasets were tested for significant overlap using Fisher's exact tests (separately for positive and negative DF). Since no reaction shows fully consistent DF across all 12 datasets analyzed, similarly as with the DE analysis, we identified the DF reactions with high level of consistency especially in the in vivo patient datasets, such that the results may be more clinically relevant: we identified DF reactions in the consistent direction (positive/negative) in at least one of the bulk RNA-seq patient datasets and also at least one of the scRNA-seq datasets, while showing DF in the opposite direction in no more than 3 datasets, resulting in 293 positive and 349 negative DF reactions. The metabolic pathway enrichment of these DF reactions was analyzed with Fisher's exact tests, with the "subSystems" slot in the metabolic model used as pathway annotation. Note that due to the special interpretation of the sign of DF values as explained above, the GSEA used for gene expression-level analysis is not appropriate for pathway enrichment analysis on the flux level.

### Differential transport flux analysis by each metabolite species

Differential transport flux analysis on the metabolite level was first performed for each dataset. Specifically, in each group (control or virus-infected), the sampled flux values on the reaction level for all transport reactions across the cell membrane were summarized by each metabolite, taking the direction of transport (i.e. import or export) into consideration to produce the net cellular transport flux of the metabolite (positive means net import, vice versa). The metabolites with differential net transport fluxes between the control and virus-infected groups were identified as in the reaction-level DF analysis described above. The consistent DF metabolites were then identified in the same way as with the DF reactions described above.

### Analysis of the consistent flux alteration patterns in different metabolic pathways

For each of the significantly enriched metabolic pathways identified in the consistent DF reaction analysis described above, we defined the "consensus" direction of each reaction as represented by those shown in the virus-infected group from the majority (>6 out of 12) of the datasets, and also similarly for the "consensus" direction of DF for each reaction. The consensus directions of reactions and their DF were overlaid onto network diagrams of the pathways and visualized, where metabolites are represented by nodes, reactions are represented by directed (hyper) edges with edge direction corresponding to the consensus reaction direction and edge color corresponding to the consensus DF direction. Parts of the metabolic pathways where reactions are not consistently altered across datasets are greyed out or removed to increase the clarity. Upon visual inspection, potential futile loops in the network are also removed from the visualizations.

### Prediction of anti-SARS-CoV-2 target metabolic reactions with metabolic transformation algorithm

For each of the collected datasets, the DE result of virus-infected vs control samples as well as the representative flux distribution of the virus-infected group computed with iMAT [@shlomi_network-based_2008] followed by ACHR sampling were used as inputs for the GEM-based metabolic transformation algorithm (MTA) [described in Chapter \@ref(mta), based on @yizhak_model-based_2013; @valcarcel_rmta_2019) to predict metabolic reactions whose knock-out can transform cellular metabolic state from that of the virus-infected to that of the control samples. The output of MTA is a score (MTA score) for each metabolic reaction, with higher scores corresponding to better candidates for achieving the metabolic transformation as specified above. From our previous experience [@yizhak_model-based_2013], the top 10-20% MTA predictions contain promising targets. The human Recon 1 GEM [@duarte_global_2007] was used for the MTA analysis. The MTA algorithm implemented in our in-house R package named gembox was used, with the academic version of IBM ILOG CPLEX Optimization Studio 12.10 as the optimization solver. To compare the MTA predictions across datasets, the top 10% predictions from each pair of datasets were tested for significant overlap using Fisher's exact tests.

### Validation of the MTA-predicted anti-SARS-CoV-2 metabolic targets

Multiple datasets of reported anti-SARS-CoV-2 gene targets or drugs identified from large-scale chemical or genetic screens were collected to validate our predictions. Gene-level results of two published CRISPR-Cas9 genetic screens [@wei_genome-wide_2021; @daniloski_identification_2021] were obtained from the supplementary materials of the respective publication. For [@wei_genome-wide_2021], gene hits with FDR<0.1 and mean z score>0 (i.e. KO inhibits the viral infection) were taken; [@daniloski_identification_2021] reported two screens with different multiplicities of infections (MOIs) and provided only single-sided FDR, so gene hits with FDR<0.1 from either screen were taken. The union set of hits from both studies were used. Lists of experimentally validated drugs reported in different studies compiled by [@kuleshov_covid-19_2020] were downloaded from https://maayanlab.cloud/covid19/, which are then mapped to the genes they inhibit using data from DrugBank v5.1.7 [@wishart_drugbank_2018]. Additionally, host proteins identified to interact with SARS-CoV-2 proteins were obtained from the supplementary materials of [@gordon_sars-cov-2_2020] and [@stukalov_multi-level_2020]. The genes from these validation datasets are mapped to metabolic reactions wherever applicable based on the human GEM Recon 1 [@duarte_global_2007] data. Then, the significant overlap between the top 10% MTA-predicted targets from each dataset and each of the validation sets described above was tested with Fisher's exact tests on the reaction level. Reaction-level test is performed because multiple reactions can be mapped to the same gene, and performing Fisher's exact test on the gene-level fails to consider such multiple mapping and is thus inappropriate.

For ROC analysis, negative sets (i.e. genes whose KO promotes SARS-CoV-2 infection) were defined based on the two CRISPR-Cas9 screens described above. For [@wei_genome-wide_2021], gene with FDR<0.1 and mean z score<0 were taken; since [@daniloski_identification_2021] provided only single-sided FDR, the log fold-change threshold corresponding to the FDR<0.1 cutoff was identified, and genes with more extreme log fold-changes in the opposite direction were taken. The union of the negative sets from both studies was used. Both the positive (described in the previous paragraph) and negative sets of genes are then mapped to metabolic reactions as described above. The negative set defined as such contains a relatively balanced number of reactions compared to the positive set (131 vs 81). The score for the reactions produced by MTA was used as the predicted value for ROC analysis. The R package pROC [@robin_proc_2011] was used to compute the AUROC values and their 95% confidence intervals (the latter computed with bootstrapping).

### Defining and analyzing the consensus set of candidate anti-SARS-CoV-2 metabolic targets across datasets

Based on top 10% MTA predictions from the 12 datasets (6 in vitro and 6 in vivo), the metabolic reaction targets that are recurrent in at least 3 of the in vitro datasets, and also in 3 of the in vivo datasets (i.e. the intersection of the two) were taken to be the final consensus candidate targets with high-confidence support across datasets. These target reactions are also mapped to genes using the human GEM Recon 1 data [@duarte_global_2007], and further mapped to known drugs inhibiting the gene targets using data from DrugBank v5.1.7 [@wishart_drugbank_2018]. The metabolic pathway enrichment of these targets was analyzed with Fisher's exact tests, with the "subSystems" slot in the metabolic model used as pathway annotation. For enrichment analysis of cellular transport reactions by each metabolite species, the reactions responsible for transporting each metabolite across the cell membrane were identified from Recon 1 and used as reaction sets for the Fisher's exact tests.

### Preparation of Vero E6 cell samples with SARS-CoV-2 infection and remdesivir treatment, RNA-sequencing, and gene expression data analysis

Vero E6 cells (ATCC® CRL-1586™) were maintained in Dulbecco's modified eagle medium (DMEM, Gibco) supplemented with 10% heat-inactivated fetal bovine serum (FBS, Gibco), 50 U ml−1 penicillin, 50 μg ml−1 streptomycin, 1 mM sodium pyruvate (Gibco), 10 mM 4-(2-hydroxyethyl)-1-piperazineethanesulfonic acid (HEPES, Gibco), and 1× MEM non-essential amino acids solution (Gibco). The SARS-CoV-2 USA-WA1/2020 strain was obtained from BEI Resources (NR-52281). The virus was inoculated on Vero E6 cells and the cell supernatant was collected at 72 h post-inoculation (hpi), when extensive cytopathic effects were observed. The supernatant, after clarification by centrifugation 15 min at 4 °C at 5,000 xg, was aliquoted and stored at −80 °C until use. 500,000 Vero E6 cells were seeded in 6-well plates. The following day, the cell medium was replaced with fresh medium supplemented with either DMSO or 1 µM remdesivir (Adooq Bioscience), and cells were either mock-infected or infected with SARS-CoV-2 USA-WA1/2020 (MOI=0.3). Twenty-four hours after infection, cells were collected, and total intracellular RNA was extracted using the Qiagen® RNeasy® Plus Mini Kit. Three replicates were performed for each group, resulting in a total of six samples. The quality of the extracted RNA was assessed with Agilent® 2100 Bioanalyzer. Libraries were prepared on total RNA following ribosome RNA depletion with standard protocol according to Illumina®. Total RNA sequencing was then performed on the Illumina® NextSeq system, 150bp paired-end runs were performed and 100 million raw reads per sample were generated. STAR [@dobin_star_2013] was used to align the reads to reference genome of the African green monkey (Chlorocebus sabaeus, https://useast.ensembl.org/Chlorocebus_sabaeus/Info/Annotation), with the SARS-CoV-2 genome (https://www.ncbi.nlm.nih.gov/nuccore/NC_045512) added to the reference genome. DESeq2 [@love_moderated_2014] was used for DE analysis between pairs of experimental groups (including virus+remdesivir vs control and virus vs control). GSEA [@subramanian_gene_2005] was performed using the implementation in the R package fgsea [@korotkevich_fast_2021]. The gene set/pathway annotations used were the Reactome [@jassal_reactome_2020] and KEGG [@kanehisa_kegg_2021] subsets from the "Canonical Pathway" category in version 7.0 MSigDB database [@liberzon_molecular_2011].

### Genome-scale metabolic modeling of the remdesivir-treated Vero E6 cell samples and prediction of anti-SARS-CoV-2 metabolic targets in combination with remdesivir

As with the metabolic modeling of the other datasets on SARS-CoV-2 infection, iMAT [@shlomi_network-based_2008] together with ACHR was used to compute the metabolic flux distribution for each of the experimental groups, using the median expression TPM values of each group as the input to iMAT. Reactions with differential fluxes (DF) between groups (including virus+remdesivir vs control and virus vs control) were identified as described above, and their significant metabolic pathway enrichment was tested with Fisher's exact tests, with pathways defined by the "subSystems" from the Recon 1 model [@duarte_global_2007]. The DE result of virus+remdesivir vs control group and the mean flux distribution of the virus+remdesivir group computed with iMAT were used as inputs for MTA to predict metabolic reactions whose knock-out can further transform the virus+remdesivir metabolic state to the normal control state. The top 10% and 20% MTA-predicted targets were tested for significant enrichment for the targets of a list of experimentally validated synergistic drugs with remdesivir [@nguyenla_discovery_2020] using Fisher's exact test (performed on the metabolic reaction level as described above). Metabolic pathway enrichment analysis of the top MTA-predicted targets was performed as described above.

### Softwares and code

R version 3.6.3 was used for all statistical tests. P values lesser than 2.22e-16 may not be computed accurately and are reported as "P<2.22e-16" throughout the text. The Benjamini-Hochberg (BH) method was used for P value adjustment throughout the text. The R packages ggplot2 [@wickham_ggplot2_2009], ComplexHeatmap [@gu_complex_2016] and visNetwork (https://cran.r-project.org/web/packages/visNetwork/index.html) were used to create the visualizations. The code used for the analyses can be found in the GitHub repository: https://github.com/ruppinlab/covid_metabolism.

## Discussion and Conclusion

In this study, I took advantage of multiple published studies involving the gene expression profiling of SARS-CoV-2 infected samples, in order to comprehensively reanalyze these existing dataset with both gene expression-level and metabolic flux-level methods and discover recurrent virus-associated changes. The published studies involve a variety of different experimental models and/or sample types, and although I was not able to collect and include all available datasets given their vast numbers, the selection of datasets was to cover those from both popular *in vitro* cell-based models of SARS-CoV-2 infection as well as human patients (nasopharyngeal swab and BALF samples). Another consideration is to cover datasets using different platforms. Specifically, I included a few scRNA-seq data to separate the distinctive changes within the epithelial cells from, e.g. various types of immune cells. In the current study, the main focus was on the epithelial cells, representing the major cell type infected by SARS-CoV-2, although it is also highly desirable to systematically investigate the SARS-CoV-2 induced changes (including metabolic changes) in the other cell types especially different immune cells.

For the method used for the combined analysis of multiple datasets, I have avoided a more formal effect size-based meta-analysis due to the challenge arising from the wide range of expression profiling platforms (bulk RNA-seq, scRNA-seq, MS-based proteomics) and the technical variation involved. Instead, in comparing the molecular alterations identified from across the datsets, I used the approach of comparing the significant changes (on the gene, pathway or flux level) and testing for their similarities with Fisher's exact tests, which helped to mitigate the variation across datasets due to different platforms and techinal noise. In the identification of robust alterations supported by multiple datasets, due to the same reason as discussed above, a formal effect size-based analysis is challenging, and approaches for combining P values, for example the Fisher's method can be more practical. In this study, prior knowledge and subjected criteria were involved in the procedure which assigned more weights to the significant findings (genes/pathways/reactions) from the *in vivo* patient datasets, as these may have higher clinical relevance.

Using the above integrated approach, the results show that different pathways of immune responses, including those response to virus infection, are recurrently upregulated upon SARS-CoV-2 infection across datasets (Figure \@ref(fig:covid-de-gsea-consensus)), as expected. Antigen presentation via MHC-I is consistently downregulated upon the infection, consistent with the report that the ORF8 protein of SARS-CoV-2 can effectively downregulate MHC-I [@zhang_orf8_2020]. The downregulation of antigen presentation has long been observed during the infection of many other viruses as well, as a mechanism of the viruses to evade immune detection [@hewitt_mhc_2003]. Besides immune responses, more intriguingly, metabolism is one of the cellular domains that exhibit the most coherent changes across datasets in SARS-CoV-2 infection, which echoes the previous reports that many viruses need to manipulate host metabolism for their proliferation and intracellular life cycle [@mayer_hijacking_2019; @thaker_viral_2019]. While it appears that most of the consistent metabolic changes involves downregulated metabolic genes (Figure \@ref(fig:covid-de-gsea-consensus)), it is not true that SARS-CoV-2 infection resulted in globally decreased metabolic activities, as can be seen in Appendix Figure \@ref(fig:covid-gsea-metab). Besides, the pathway level changes should be carefully inspected based on the genes driving the enrichment to avoid misinterpretation of results. For example, the pathway "TCA cycle and electron transport chain" is seen consistently downregulated after SARS-CoV-2 infection, but upon closer inspection it's only the ETC genes that are downregulated, while TCA cycle genes are not. Actually, from the GEM analysis it can be seem that the TCA cycle mostly has upregulated fluxes. 

Therefore, it can also be seen that GEM can be a valuable complementary approach to gene expression-level analysis. It is known that gene expression does not always perfectly correlate with protein level or enzyme activity [@maier_correlation_2009]. Besides, many metabolic reactions are reversible, while the directions of reactions are important biologically, such information is missing on the gene level. With GEM, the computed metabolic fluxes can reveal the extra complexity in SARS-CoV-2-induced metabolic reprogramming. Many of the metabolic flux changes computed by GEM modeling are consistent with prior knowledge on SARS-CoV-2 and other related viruses. For example, the highly coherent increase in pyrimidine biosynthesis corresponds to the increased need of viral genome replication and gene expression [@bojkova_targeting_2020], and pyrimidine de novo synthesis inhibitors have been shown to have anti-SARS-CoV-2 effects [@xiong_novel_2020]. Inositol phosphate metabolism is important for the life cycle of many viruses due to the structural or signaling roles of different phosphoinositides [@beziau_role_2020], with the inhibition of certain phosphoinositides disrupting endocytosis and blocking SARS-CoV-2 cell-entry [@ou_characterization_2020]. Fatty acid synthesis was reported to increase in SARS-CoV-2 infection [@ehrlich_sars-cov-2_2020]; our results reveal a potentially more complex pattern for different types of fatty acids, which is paralleled by the results from several metabolomics studies [@shen_proteomic_2020; @barberis_large-scale_2020; @thomas_covid-19_2020]. Both TCA cycle and OXPHOS have been shown to decrease based on gene expression during the virus infection [@ehrlich_sars-cov-2_2020; @gardinassi_immune_2020], and have been implicated in the systemic syndromes of the virus [@li_metabolic_2021]. In the GEM analysis of this study, the results only confirmed the decrease of OXPHOS but revealed a mostly normal-functioning TCA cycle on the flux level, which appears to be driven by 2-oxoglutarate transported from the cytosol (Appendix Figures \@ref(fig:covid-flux-tca1) and \@ref(fig:covid-flux-tca2)). These flux-level patterns computed by GEM modeling should be verified with isotope labeling experiments, and their biological significance in the virus infection needs to be further investigated.

Given the importance of metabolism during virus infection, targeting host metabolism has already been proposed as a promising novel antiviral strategy in general [@mayer_hijacking_2019]. Based on this proposal, I applied the MTA framework to predict anti-SARS-CoV-2 targets that act via reverting the virus-induced metabolic perturbations. MTA is not based on supervised machine learning technique, and do not use any of the validation datasets for target prediction. Yet, the MTA predictions showed good performance during validation. It is particularly encouraging to see that in several cases, the validation data which was mostly from *in vitro* experiments correlated well with the predictions based on *in vivo* human data, for example, the predictions from the human patient scRNA-seq dataset from [@liao_single-cell_2020] produced AUROC of above 0.75 when validated against the CRISPR-Cas9 screen data from *in vitro* cell lines. In some datasets, however, our predictions were not successfully validated by the genetic screen data (Figures \@ref(fig:covid-mta-genetic-enr) and \@ref(fig:covid-mta-auroc)), one reason could be that MTA can only consider the metabolism-related effects and ignores other potential mechanisms that determine the antiviral efficacy of a target. It could also be due to biological differences between the datasets used for prediction and those used for validation. I therefore did not explicitly exclude the datasets with suboptimal performances in the validation to avoid overdependence on the limited validation sets when defining the most promising set of candidates. Instead, I adopted a criteria of maximal support from as many datasets as possible while assigning more weights to the *in vivo* human patient datasets to achieve higher clinical relevance (as explained in Methods). In terms of the prediction for combinatory targets with remdesivir, the preliminary validation showed promising results despite that validation is much more limited due to lack of data. Follow-up studies are required to further validate the antiviral efficacy of these predicted targets.

In conclusion, it was discovered that metabolic reprogramming of the host cells is a robust characteristics of SARS-CoV-2 infection, via the combined analysis of multiple gene expression datasets on SARS-CoV-2 infection using both common gene-level bioinformatic analysis as well as metabolic modeling. The modeling approached added to the gene-level analysis in revealing fine-grained metabolic alteration patterns in the metabolic network, and also was critical in the prediction of metabolic anti-SARS-CoV-2 targets (both single and combinatory with remdesivir). Although the predition was shown to have good performance based on validation agains reported experimentally validated gene targets and drugs, these predictions should be further validated with rigorous experiments.

  
