<!--I commented out this header
`r if(knitr:::is_latex_output()) '\\appendix'`

`r if(!knitr:::is_latex_output()) '# (APPENDIX) Appendix {-}'` 
-->

```{r, include=FALSE}
library(my.utils)
library(gembox)
library(ComplexHeatmap)
library(circlize)
library(cowplot)
```

# Appendices {-}

Additional figures and tables for each chapter are given below.

## Chapter \@ref(mta) {-}

```{r}
tmp <- readRDS("data/mta/datasets.rds")
```

```{r mta-datasets, results="asis"}
knitr::kable(tmp, format="latex",
  caption = "Gene expression datasets involving metabolic gene perturbation (KO/KD/protein inhibition) collected for the validation of the MTA algorithm and its variants",
  longtable = TRUE,
  booktabs = TRUE
)# %>% kableExtra::kable_styling(latex_options="scale_down")
```

```{r mta-rmta-vs-moma-and-mmta, fig.cap="Comparison of the predicted percentage ranks of the ground truth reactions across datasets from rMTA, mMTA and MOMA, with the best parameter combinations.", fig.width=4.5, fig.height=2}
rmta <- new.env()
load("data/mta/rmta.check.res.RData", envir=rmta)
a <- qplot(x=rmta$res$rmta["1e-04","0.1","10",]*100, y=rmta$res$moma["1e-04","0.1","10",]*100, xlab="rMTA %rank", ylab="MOMA %rank") + theme_classic()
b <- qplot(x=rmta$res$rmta["1e-04","0.1","10",]*100, y=rmta$res$mmta["1e-04","0.98","10",]*100, xlab="rMTA %rank", ylab="mMTA %rank") + theme_classic()
plot_grid(a, b, nrow=1)
```

```{r mta-rmta-de-rm-ctrl, fig.cap="The prediction performance of rMTA across datasets with various parameters, after removing the ground truth gene from the DE result given as the input to rMTA.", fig.width=5, fig.height=7}
rmta.rm <- new.env()
load("data/mta/rmta.rm.true.gene.res.RData", envir=rmta.rm)

colf <- colorRamp2(c(5,15,30), c("grey95","orange","red"))
tmpf <- function(i) {
  mat <- rmta.rm$summ.cnt$rmta[[i]][6:1,]
  hm <- Heatmap(mat, 
        name="#datasets\n ground truth\n among top 10%",
        row_title=sprintf("alpha=%s\n\n epsilon",i),
        row_names_side="left",
        cluster_rows=FALSE,
        column_title=NULL,
        column_title_side="bottom",
        column_names_side="bottom",
        cluster_columns=FALSE,
        col=colf,
        cell_fun=function(j, i, x, y, width, height, fill) grid.text(sprintf("%d", mat[i,j]), x, y, gp=gpar(fontsize=10))
        )
}

hm <- tmpf(names(rmta.rm$summ.cnt$rmta)[length(rmta.rm$summ.cnt$rmta)])
for (i in rev(names(rmta.rm$summ.cnt$rmta)[-length(rmta.rm$summ.cnt$rmta)])) {
  hm <- hm %v% tmpf(i)
}

draw(hm, column_title="n", column_title_side="bottom")
```

```{r mta-mmetal-de-rm-ctrl, fig.cap="The prediction performance of mMeTAL across datasets with various parameters, after removing the ground truth gene from the DE result given as the input to mMeTAL.", fig.width=5, fig.height=2}
rmetal.rm <- new.env()
load("data/mta/rmetal.rm.true.gene.res.RData", envir=rmetal.rm)

mat <- rmetal.rm$summ.cnt$mmetal.1
colf <- colorRamp2(c(0,15,30), c("grey95","orange","red"))
Heatmap(mat, 
        col=colf,
        name="#datasets\n ground truth\n among top 10%",
        row_title="alpha",
        row_names_side="left",
        cluster_rows=FALSE,
        column_title="n",
        column_title_side="bottom",
        column_names_side="bottom",
        cluster_columns=FALSE,
        cell_fun=function(j, i, x, y, width, height, fill) grid.text(sprintf("%d", mat[i,j]), x, y, gp=gpar(fontsize=10))
)
```

```{r mta-moma-de-rm-ctrl, fig.cap="The prediction performance of MOMA across datasets with various parameters, after removing the ground truth gene from the DE result given as the input to MOMA.", fig.width=5, fig.height=1.5}
mat <- rmta.rm$summ.cnt$moma[[1]][1,,drop=FALSE]
rownames(mat) <- NULL
colf <- colorRamp2(c(5,15,30), c("grey95","orange","red"))
Heatmap(mat, 
        col=colf,
        name="#datasets\n ground truth\n among top 10%",
        row_title=NULL,
        #row_names_side="left",
        cluster_rows=FALSE,
        column_title="n",
        column_title_side="bottom",
        column_names_side="bottom",
        cluster_columns=FALSE,
        cell_fun=function(j, i, x, y, width, height, fill) grid.text(sprintf("%d", mat[i,j]), x, y, gp=gpar(fontsize=10))
)
```

## Chapter \@ref(ucp2) {-}

```{r ucp2-lu-df-oxphos, fig.cap="Computed metabolic flux distributions of the ETC/OXPHOS reactions in the 9.2-P persistent T cells vs the 9.1-NP non-persistent T cells in ACT.", fig.width=6, fig.height=2}
imat.res.lu <- readRDS("data/ucp2/Lu/imat.res.RDS")
plot.fluxes2(imat.res.lu$np9$result.model, imat.res.lu$p9$result.model, group.names=c("non-persistent (clone 9.1-NP)", "persistent (clone 9.2-P)"),
             rxns=list(`Complext 1`="NADH2_DASH_u10m", `Complext 3`="CYOR_DASH_u10m", `Complext 4`="CYOOm3", `ATP Synthase`="ATPS4m"),
             ylims=list(c(-10,40), c(-10,40), c(-10,40), c(20,100)))
```

```{r ucp-cart-mta-mitotx-gsea, fig.cap="GSEA plot showing the enrichment for the mitochonrial transport pathway by the top MTA predictions on essential reactions to T cell anti-cancer function in CAR-T therapy.", fig.width=3, fig.height=1.6}
load("data/ucp2/Fraietta/mta.res.RData")
rs.tx.mito <- recon1$rxns[recon1$subSystems=="Transport, Mitochondrial"]
x <- mta.res[, score-score[id==0]]
names(x) <- mta.res$reaction
x <- x[!is.na(x)]
fgsea::plotEnrichment(rs.tx.mito, x)
```

```{r ucp-cart-mta-mitoh-gsea, fig.cap="GSEA plot showing the enrichment for the mitochonrial proton transport reactions by the top MTA predictions on essential reactions to T cell anti-cancer function in CAR-T therapy.", fig.width=3, fig.height=1.6}
rs.h <- get.transport.info(recon1, "h", "m", "c")$h$rxn
fgsea::plotEnrichment(rs.h, x)
```

```{r}
load("data/ucp2/Lu/check.mta.res.RData")
gs.mito.tx <- get.transport.info(recon1, c1="m", c2="c")
gs.mito.tx <- lapply(gs.mito.tx, function(x) x$rxn)
tmp <- rbind(path.enr.res10$p2np[reaction.set=="Transport, Mitochondrial"],
             enrich.gsets(top10.mta.hits$p2np$reaction, gs.mito.tx, recon1$rxns, name="reaction")[reaction.set=="h"][, reaction.set:="Mitochondrial Proton Transport"])[, -"overlap.reactions"]
```

```{r ucp2-lu-mta-mitotx-h, results="asis"}
knitr::kable(tmp, format="latex",
  caption = "Enrichment of mitochondrial transport pathway and mitochondrial proton transport reactions by the top 10% MTA predicted reactions whose KO can reduce T cell persistence in ACT therapy.",
  #longtable = TRUE,
  booktabs = TRUE
) %>% kableExtra::kable_styling(latex_options="scale_down")
```

```{r ucp2-lu-df-ucp, fig.cap="Mitochondrial uncoupling reaction rate in the 9.2-P persistent T cells vs the 9.1-NP non-persistent T cells in ACT.", fig.width=1.5, fig.height=1.5}
plot.fluxes2(imat.res.lu$np9$result.model, imat.res.lu$p9$result.model, group.names=c("non-persistent (clone 9.1-NP)", "persistent (clone 9.2-P)"), rxns="Htm", ylab="Mitochondrial Uncoupling\n Reaction Rate", ylim=c(NA,10)) +
  theme(axis.title.y=element_text(size=10),
        legend.direction="vertical",
        legend.box.margin=margin(-8,5,-8,-20))
```

```{r ucp2-icb-ucp2-de, out.width="100%", fig.cap="UCP2 expression levels in the tumor samples from responders and non-responders in various ICB datasets."}
knitr::include_graphics(path="figure/ucp2/icb.ucp2.de.png")
```

```{r ucp2-genipin-fc, out.width="100%", fig.cap="Flow cytometry results on the level of various T cell activation, effector function, memory and senescence gene markers in genipin-treated vs control mice Pmel-1 CD8 T cells. The gene markers are as follows: T cell activation (CD25, CD44), memory (CD62L), effector molecule (GzmB), senescence (KLRG1) and exhaustion (PD1). Figure credit: Dr. Madhusudhanan Sukumar"}
knitr::include_graphics(path="figure/ucp2/genipin.markers.fc.png")
```

## Chapter \@ref(covid) {-}

```{r covid-gsea-mat, out.width="80%", fig.cap="Pairwise comparison of the datasets by their top significantly enriched pathways with Fisher's exact tests."}
knitr::include_graphics(path="figure/covid/gsea.enr.mat.svg.png")
```

```{r covid-gsea-metab, out.width="100%", fig.cap="Summary of the metabolic pathway changes from GSEA analysis across the analyzed datasets."}
knitr::include_graphics(path="figure/covid/metab.pathways.svg.png")
```

```{r covid-df-pos-mat, out.width="80%", fig.cap="Pairwise comparison of the datasets by their top positive DF reactions with Fisher's exact tests."}
knitr::include_graphics(path="figure/covid/df.mat.pos.svg.png")
```

```{r covid-df-neg-mat, out.width="80%", fig.cap="Pairwise comparison of the datasets by their top negative DF reactions with Fisher's exact tests."}
knitr::include_graphics(path="figure/covid/df.mat.neg.svg.png")
```

```{r covid-flux-tca1, out.width="80%", fig.cap="Visualization of the consistent flux changes in the TCA cycle."}
knitr::include_graphics(path="figure/covid/tca1.png")
```

```{r covid-flux-tca2, out.width="80%", fig.cap="Visualization of the consistent flux changes in the reactions involving the cytosolic counterparts of TCA cycle metabolites."}
knitr::include_graphics(path="figure/covid/tca2.png")
```

```{r covid-flux-inositol, out.width="80%", fig.cap="Visualization of the consistent flux changes in the inositol phosphate metabolism pathway."}
knitr::include_graphics(path="figure/covid/ip.png")
```

```{r covid-flux-glycine, out.width="80%", fig.cap="Visualization of the consistent flux changes in the glycine, serine, and threonine metabolism pathway."}
knitr::include_graphics(path="figure/covid/glyser.png")
```

```{r covid-mta-ppi-enr, out.width="100%", fig.cap="Summary of the enrichment of top MTA predictions from each dataset in the host-virus PPI proteins."}
knitr::include_graphics(path="figure/covid/mta.ppi.enr.svg.png")
```

```{r covid-mta-roc-calu3, out.width="80%", fig.cap="An example ROC curve for the Calu-3 dataset."}
knitr::include_graphics(path="figure/covid/auc.calu.svg.png")
```

```{r covid-mta-roc-liao, out.width="80%", fig.cap="An example ROC curve for the Liao et al. single-cell dataset."}
knitr::include_graphics(path="figure/covid/auc.liao.svg.png")
```

```{r covid-rem-expr-pathway, out.width="100%", fig.cap="Summary of the significant pathways with differentially expressed genes between the virus+remdesivir and the control Vero E6 cells"}
knitr::include_graphics(path="figure/covid/enr.c2vd.expr.svg.png")
```

```{r covid-rem-flux-pca, out.width="80%", fig.cap="PCA plot on the iMAT-computed metabolic fluxes from control Vero E6 cells as well as SARS-CoV-2-infected cells with or without remdesivir treatment"}
knitr::include_graphics(path="figure/covid/rem.pca.flux.svg.png")
```
